---
title: "Resuscitating the microbial seed bank alters plant-soil interactions"
author: "Venus Kuo"
date: "June 25, 2018"
output: html_document
---


Questions: 

1) Does resuscitating the soil microbial seed bank with Rpf alter plant biomass and reproductive output?

2) How does Rpf alter plant-soil microbe interactions?


Methods:

1) Conducted 6-week growth chamber experiment using Brassica rapa grown in live field soil treated with either weekly application of recombinant Rpf or control protein buffer solution. 

2) Measured weekly soil CO2 respiration from experimental pots using 1 gram of soil incubated for 24 h. 

3) Measured plant traits at the end of the exeriment. 

4) Quantified soil bacterial and fungal abundance using qPCR of the 16S rRNA and ITS gene, respectively, from week 6 of the experiment.

5) Determined Rpf and metabolic status effects on the soil bacterial community structure by sequencing soil bacterial 16S rRNA using MiSeq Illumina and performing alpha- and beta-diversity analyses.  


# Set working environment and load packages 

```{r setup, message = FALSE, warning = FALSE}
# For analyses for figures 2 to 4 
# Clear and set working directory 
rm(list = ls())
getwd()
knitr::opts_knit$set(root.dir = "~/../Github/BrassicaRpf/data")

# Require or install packages 
package.list <- c('vegan', 'nlme' ,'data.table', 'plyr', 'pander', 'reshape', 'grid', 'png', 'car', 'bbmle', 'reshape2', 'ggplot2') 
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
  } }

# Load small sem calculation function 
sem <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
```


# 1) Rpf effects on Brassica rapa traits 

```{r}
# Load plant trait data 
plant.data <- read.delim("plantTrait.txt", sep = ",", header = TRUE)
# Remove soil sterilization treatment from dataset # 
plant.data <- subset(plant.data, soil == "Live") 

# Subseting data for relevant information 
data.sub <- subset(plant.data, select=(c(Treatment, abiomass, bbiomass)))
# Combining the above and belowground biomass # 
data.m <- melt(data.sub) 

# Above-ground shoot biomass
# Selecting shoot biomass data for plotting # 
shoot.rpf <- data.m[ which(data.m$Treatment == "Rpf+" & data.m$variable == "abiomass"),]
shoot.con <- data.m[ which(data.m$Treatment == "Rpf-" & data.m$variable == "abiomass"),]
# Generating shoot biomass mean and sem table # 
shoot.mean <- aggregate(plant.data$abiomass ~ Treatment, plant.data, mean)
shoot.sem <- aggregate(plant.data$abiomass ~ Treatment, plant.data, sem)
shoot.sem.LL <- shoot.mean[2] + shoot.sem[2]
shoot.sem.UL <- shoot.mean[2] - shoot.sem[2]
shoot.table <- data.frame(shoot.mean[1], shoot.mean[2], shoot.sem[2],
          shoot.sem.LL[1], shoot.sem.UL[1])
colnames(shoot.table) <- c("Treatment", "mean", "sem", "LCI", "UCI")
shoot.table <- shoot.table[order(shoot.table$mean),]

# Below-ground root biomass 
# Selecting root biomass data for plotting # 
root.rpf <- data.m[ which(data.m$Treatment == "Rpf+" & data.m$variable == "bbiomass"),]
root.con <- data.m[ which(data.m$Treatment == "Rpf-" & data.m$variable == "bbiomass"),]
# Generating root biomass  mean and sem table # 
root.mean <- aggregate(plant.data$bbiomass ~ Treatment, plant.data, mean)
root.sem <- aggregate(plant.data$bbiomass ~ Treatment, plant.data, sem)
root.sem.LL <- root.mean[2] + root.sem[2]
root.sem.UL <- root.mean[2] - root.sem[2]
root.table <- data.frame(root.mean[1], root.mean[2], root.sem[2],
          root.sem.LL[1], root.sem.UL[1])
colnames(root.table) <- c("Treatment", "mean", "sem", "LCI", "UCI")
root.table <- root.table[order(root.table$mean),]

# Flower number per plant
# Flower count data for plotting # 
flower.rpf <- plant.data[ which(plant.data$Treatment == "Rpf+"),]
flower.con <- plant.data[ which(plant.data$Treatment == "Rpf-"),]
# Generate flower count data table # 
flower.mean <- aggregate(plant.data$flower.count ~ Treatment, plant.data, mean)
flower.sem <- aggregate(plant.data$flower.count ~ Treatment, plant.data, sem)
flower.sem.LL <- flower.mean[2] + flower.sem[2]
flower.sem.UL <- flower.mean[2] - flower.sem[2]
flower.table <- data.frame(flower.mean[1], flower.mean[2], flower.sem[2],
          flower.sem.LL[1], flower.sem.UL[1])
colnames(flower.table) <- c("Treatment", "mean", "sem", "LCI", "UCI")
flower.table <- flower.table[order(flower.table$mean),]


# Generating 3 panel figure showing Rpf effects on plant traits # 
png(filename="../figures/Figure1-PlantTraits.png",
    width = 1200, height = 800, res = 96*2)

par(oma=c(7,3,7,1), mar=c(2,3,3,3.5), mfrow=c(1,3))

# Plotting first panel for shoot biomass # 
shoot.fig <- plot(jitter(rep(1, length(shoot.con$value)), amount = 0.1), shoot.con$value, 
      ylim = c(0, 1.5), xlim = c(0.5, 2.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 3.5,
      cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 2,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
points(jitter(rep(2, length(shoot.rpf$value)), amount = 0.1), shoot.rpf$value, pch = 21, 
       bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

# Adding mean value point # 
points(1, mean(shoot.con$value), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  
points(2, mean(shoot.rpf$value), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)

box(lwd = 2)

# Y-axis label # 
mtext(expression('Shoot biomass (g)'), side = 2,
      outer = FALSE, cex = 1.25, line = 2.5, adj = 0.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0", "0.5", "1.0", "1.5"), at = c(0, 0.5, 1.0, 1.5))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at=c(0, 0.5, 1.0, 1.5), labels = F, tck = -0.05)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("Rpf-", "Rpf+"), at = c(1, 2))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at = c(1, 2), labels = F, tck = -0.05)

# Adding SEM bars # 
arrows(x0 = c(2,1), y0 = shoot.table$mean, y1 = shoot.table$LCI, angle = 90,
       length = 0.15, lwd = 2)
arrows(x0 = c(2,1), y0 = shoot.table$mean, y1 = shoot.table$UCI, angle = 90,
       length=0.15, lwd = 2)

# Panel label # 
text(0.65,1.45 ,labels="A", col="black", cex=2)

# p-value label 
mtext(text = expression(italic("P")~" = 0.017") , side =3, line = -1.2, adj = 0.925, col="black", cex=0.8)


# Plotting second panel for root biomass # 
root.fig <- plot(jitter(rep(1, length(root.con$value)), amount = 0.1), root.con$value, 
      ylim = c(0, 1.5), xlim = c(0.5, 2.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 3.5,
      cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 2,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
points(jitter(rep(2, length(root.rpf$value)), amount = 0.1), root.rpf$value, pch = 21, 
       bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

# Adding mean circle # 
points(1, mean(root.con$value), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  
points(2, mean(root.rpf$value), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)

box(lwd = 2)

# Y-axis label # 
mtext(expression('Root biomass (g)'), side = 2,
      outer = FALSE, cex = 1.25, line = 2.5, adj = 0.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0", "0.5", "1.0", "1.5"), at = c(0, 0.5, 1.0, 1.5))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at=c(0, 0.5, 1.0, 1.5), labels = F, tck = -0.05)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("Rpf-", "Rpf+"), at = c(1, 2))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at = c(1, 2), labels = F, tck = -0.05)

# Adding SEM bars # 
arrows(x0 = c(2,1), y0 = root.table$mean, y1 = root.table$LCI, angle = 90,
       length = 0.15, lwd = 2)
arrows(x0 = c(2,1), y0 = root.table$mean, y1 = root.table$UCI, angle = 90,
       length=0.15, lwd = 2)

# Panel label 
text(0.65,1.45 ,labels="B", col="black", cex=2)

# p-value label 
mtext(text = expression(italic("P")~" = 0.049") , side =3, line = -1.2, adj = 0.925, col="black", cex=0.8)


# Plotting third panel for flower count # 
flowercount.fig <- plot(jitter(rep(1, length(flower.con$flower.count)), amount = 0.1), flower.con$flower.count, 
      ylim = c(0, 15), xlim = c(0.5, 2.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 3.5,
      cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 2,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
points(jitter(rep(2, length(flower.rpf$flower.count)), amount = 0.1), flower.rpf$flower.count, pch = 21, 
       bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

points(1, mean(flower.con$flower.count), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(2, mean(flower.rpf$flower.count), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  

box(lwd = 2)

# Y axis label # 
mtext(expression('Flower number'), side = 2,
      outer = FALSE, cex = 1.25, line = 2.5, adj = 0.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0", "5", "10", "15"), at = c(0, 5, 10, 15))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at=c(0, 5, 10, 15), labels = F, tck = -0.05)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("Rpf-", "Rpf+"), at = c(1, 2))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at = c(1, 2), labels = F, tck = -0.05)

# Add SEM bars
arrows(x0 = c(2,1), y0 = flower.table$mean, y1 = flower.table$LCI, angle = 90,
       length = 0.15, lwd = 2)
arrows(x0 = c(2,1), y0 = flower.table$mean, y1 = flower.table$UCI, angle = 90,
       length=0.15, lwd = 2)

# Panel label # 
text(0.65,14.5 ,labels="C", col="black", cex=2)

# p-value label 
mtext(text = expression(italic("P")~" = 0.097") , side =3, line = -1.2, adj = 0.925, col="black", cex=0.8)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/Figure1-PlantTraits.png")
grid.raster(img)


# Statistical test: Two sample t-test of Rpf effect on plant traits 
# Total plant biomass 
total.ttest <- t.test(total.biomass ~ Treatment, data = plant.data)
total.ttest   # Significant: t = 2.8365, df = 9.2015, p = 0.0191

# Root biomass
root.ttest <- t.test(bbiomass ~ Treatment, data = plant.data)
root.ttest    # Significant: t = 2.3017, df = 8.3705, p = 0.04895

# Shoot biomass 
shoot.ttest <- t.test(abiomass ~ Treatment, data = plant.data)
shoot.ttest   # Significant: t = 2.704, df = 14, p = 0.01712

# Flower number 
flower.ttest <- t.test(flower.count ~ Treatment, data = plant.data)
flower.ttest  # Marginally significant: t = 1.7954, df = 12.288, p = 0.09721

# Seed number
seed.ttest <- t.test(seed.count ~ Treatment, data = plant.data)
seed.ttest    # Non-significant: t = 0.83156, df = 12.983, p = 0.4207

# Shoot:Root ratio 
SRratio.ttest <- t.test(shoot.root ~ Treatment, data = plant.data)
SRratio.ttest # Non-significant: t = -1.2033, df = 12.903, p = 0.2505

# Specific leaf area (SLA)
sla.ttest <- t.test(SLA ~ Treatment, data = plant.data)
sla.ttest     # Non-significant: t = 0.80287, df = 9.1288, p = 0.4424

# Shoot height
height.ttest <- t.test(height ~ Treatment, data = plant.data)
height.ttest  # Non-significant: t = 1.1102, df = 13.384, p = 0.2865
```


# 2) Rpf effects on soil microbial CO2 respiration 

```{r}
# Load soil respirtation data #
CO2 <- read.csv("GCH_CO2.txt", sep = ",", header = TRUE)

# Subsetting relevant data from CO2 dataset # 
CO2.sub <- subset(CO2, plant=="present" & hour =="24" & soil=="live", select=c(Treatment, soil, Week.1, Week.2, Week.3, Week.4, Week.5, Week.6)) 
# Change column names # 
colnames(CO2.sub) <- c("Treatment", "soil", "1","2","3","4","5","6")
# Melt dataset into three columns of treatment, soil and weeks for anaylsis
CO2.m <- melt(CO2.sub)
# Change variable column name to week 
colnames(CO2.m) <- c("Treatment", "soil", "Week", "value")
# Set week as factor 
CO2.m$Week <- as.factor(CO2.m$Week)

# Standardizing the soil respiration values by 24 hours 
CO2.m$StdCO2 <- (CO2.m$value)/24


# Calculating CO2 respiration mean and sem table # 
CO2.means.sem <- ddply(CO2.m, c("Treatment", "Week"), summarise,
                   mean=mean(StdCO2), sem=sd(StdCO2)/sqrt(length(StdCO2)))
CO2.means.sem <- transform(CO2.means.sem, lower=mean-sem, upper=mean+sem)

# The errorbars overlapped, so use position_dodge to move them horizontally
pd <- position_dodge(0)


# Generate figure 3 for Rpf effects on soil respiration # 
co <- ggplot(CO2.means.sem, aes(x=Week, y=mean, colour=Treatment, group=Treatment)) +
      geom_errorbar(aes(ymax=upper, ymin=lower), position=position_dodge(0.1), 
                data=CO2.means.sem, width = 0.5, size=1.1) +
      geom_line(aes(linetype=Treatment) ,position=pd, size=1.3) +
      geom_point(aes(shape=Treatment), position=pd, size=4) +
      scale_shape_manual(values=c(16, 16)) +
      xlab("Time (weeks)") +
      ylab(expression(Soil~respiration~(ppm~CO[2]~d^-1/~g~soil))) 

co + scale_y_continuous(limits = c(0, 1000), breaks = seq(0, 1000, 250), 
                     sec.axis = sec_axis(~ . * 1, labels = c(" "," "," ", " ", " "))) +
     theme_classic() +
     theme(axis.text.y=element_text(colour = "black", size = 18),
          axis.text.x=element_text(colour = "black", size = 20),
          axis.ticks = element_line(size = 1.25),
          axis.ticks.length = unit(.25, "cm"),
          axis.title.y = element_text(size = 15, colour = "black", margin = margin(0,10,0,0)),
          axis.title.x = element_text(size = 18, colour = "black", margin = margin(15,10,0,10)),
          panel.border = element_rect(linetype = "solid", colour = "black", size = 2, fill = NA),
          legend.position="none") +
    scale_color_manual(values=c('gray15','gray15')) + 
  annotate("text", x = 2.5, y = 750, label = "Rpf-", cex=7) + 
  annotate("text", x = 4.5, y = 250, label = "Rpf+", cex=7)

ggsave("../figures/Figure2-Soilrespiration.png", width = 20, height = 15, units = "cm")


# Statistical tests: Rpf effects on soil CO2 respiration
# RM-ANOVA 
CO2.rm <- lme(StdCO2 ~ Week*Treatment, random = ~ 1 | soil,
              correlation = corAR1(form = ~1 | soil), 
              data = CO2.m)

# Make ANOVA table #
set.caption("RMANOVA for soil CO2 respiration")
pander(anova(CO2.rm)) 
    # Significant effect of week: F5,84 = 21.6, p < 0.000
    # Significant effect of Rpf: F1,84 = 9.676, p = 0.00255
    # Non-significant interaction: F5,84 = 1.217, p = 0.3084
```


# 3) Effect of Rpf on soil bacterial and fungal abundance and F : B ratio

```{r}
# Load microbial qPCR abundance data # 
abundance <- read.csv("qPCR.txt", sep = ",", header = TRUE) 

# remove week 1 data # 
abundance <- subset(abundance, Week == "6")
# Reset data frame index # 
rownames(abundance) <- NULL

# Standardizing gene copy abundances by soil amount # 
abundance$stdGeneCopy <- (abundance$GeneCopy)/(abundance$Soil)
# Log10 transforming standardized gene copy number # 
abundance$log10stdGeneCopy <- log10(abundance$stdGeneCopy)

# Splitting abundance dataset by fungi and bacteria # 
data.sub.bac <- subset(abundance, Gene == "rRNA")
data.sub.fun <- subset(abundance, Gene == "ITS")

# Calculating fungi to bacteria ratios # 
gca.B <- data.sub.bac$stdGeneCopy
gca.F <- data.sub.fun$stdGeneCopy
abundance$FB <- (gca.F/gca.B)
ratioFB <- abundance[1:18, ]
ratioFB <- subset(ratioFB, select=(c(Treatment, Week, FB)))


# Subsetting abundance dataset 
data.sub <- subset(abundance, select=(c(Treatment, Gene, stdGeneCopy)))
# Melting the dataset 
data.m <- melt(data.sub)

# Bacterial abundance data for plotting # 
bac.rpf <- data.m[ which(data.m$Treatment == "Rpf+" & data.m$Gene == "rRNA"),]
bac.con <- data.m[ which(data.m$Treatment == "Rpf-" & data.m$Gene == "rRNA"),]
# Bacterial abundance mean and sem table # 
bac.mean <- aggregate(data.sub.bac$stdGeneCopy ~ Treatment, data.sub.bac, mean)
bac.sem <- aggregate(data.sub.bac$stdGeneCopy ~ Treatment, data.sub.bac, sem)
bac.sem.LL <- bac.mean[2] - bac.sem[2]
bac.sem.UL <- bac.mean[2] + bac.sem[2]
bac.table <- data.frame(bac.mean[1], bac.mean[2], bac.sem[2],
          bac.sem.LL[1], bac.sem.UL[1])
colnames(bac.table) <- c("Treatment", "mean", "sem", "LCI", "UCI")
bac.table <- bac.table[order(bac.table$mean),]

# Fungal abundance data for plotting # 
fun.rpf <- data.m[ which(data.m$Treatment == "Rpf+" & data.m$Gene == "ITS"),]
fun.con <- data.m[ which(data.m$Treatment == "Rpf-" & data.m$Gene == "ITS"),]
# Fungal abundance mean and sem table # 
fun.mean <- aggregate(data.sub.fun$stdGeneCopy ~ Treatment, data.sub.fun, mean)
fun.sem <- aggregate(data.sub.fun$stdGeneCopy ~ Treatment, data.sub.fun, sem)
fun.sem.LL <- fun.mean[2] - fun.sem[2]
fun.sem.UL <- fun.mean[2] + fun.sem[2]
fun.table <- data.frame(fun.mean[1], fun.mean[2], fun.sem[2],
          fun.sem.LL[1], fun.sem.UL[1])
colnames(fun.table) <- c("Treatment", "mean", "sem", "LCI", "UCI")
fun.table <- fun.table[order(fun.table$mean),]

# Fungi: Bacteria ratio data for plotting #
FB.rpf <- ratioFB[ which(ratioFB$Treatment == "Rpf+"),]
FB.con <- ratioFB[ which(ratioFB$Treatment == "Rpf-"),]
# FB: Bacteria ratio mean and sem # 
FB.mean <- aggregate(ratioFB$FB ~ Treatment, ratioFB, mean)
FB.sem <- aggregate(ratioFB$FB ~ Treatment, ratioFB, sem)
FB.sem.LL <- FB.mean[2] + FB.sem[2]
FB.sem.UL <- FB.mean[2] - FB.sem[2]
FB.table <- data.frame(FB.mean[1], FB.mean[2], FB.sem[2],
          FB.sem.LL[1], FB.sem.UL[1])
colnames(FB.table) <- c("Treatment", "mean", "sem", "LCI", "UCI")
FB.table <- FB.table[order(FB.table$mean),]


# Generating 3 panel figure for Rpf effects on soil microbial abundance 
# Plotting microbial abundance and ratio figure # 
png(filename="../figures/Figure3-MicrobialAbundanceRatio.png",
    width = 1200, height = 800, res = 96*2)
par(oma=c(7,3,7,1), mar=c(2,3,3,4), mfrow=c(1,3))

# Panel 1: Rpf effects on bacterial abundance # 
abun.bac.fig <- plot(jitter(rep(1, length(bac.con$value)), amount = 0.1), bac.con$value, 
      ylim = c(0, 2.4E6), xlim = c(0.5, 2.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 3.5,
      cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 2,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
points(jitter(rep(2, length(bac.rpf$value)), amount = 0.1), bac.rpf$value, pch = 21, 
       bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

# Adding mean data pointfor each treatment # 
points(1, mean(bac.con$value), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(2, mean(bac.rpf$value), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  

box(lwd = 2)

# Y axis labels
mtext(expression('16S rRNA gene copy/ g soil'), side = 2,
      outer = FALSE, cex = 1, line = 3.5, adj = 0.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1, las = 1, 
     labels = c("0.0E0", "8.0E5", "1.6E6", "2.4E6"), 
     at = c(0.0E0, 8.0E5, 1.6E6, 2.4E6))

axis(side = 4, lwd.ticks = 2, cex.axis = 1, las = 1,
     at=c(0.0E0, 8.0E5, 1.6E6, 2.4E6), labels = F, tck = -0.05)

axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("Rpf-", "Rpf+"), at = c(1, 2))

axis(side = 3, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at = c(1, 2), labels = F, tck = -0.05)

# Adding SEM # 
arrows(x0 = c(2,1), y0 = bac.table$mean, y1 = bac.table$LCI, angle = 90,
       length = 0.15, lwd = 2)
arrows(x0 = c(2,1), y0 = bac.table$mean, y1 = bac.table$UCI, angle = 90,
       length=0.15, lwd = 2)

# Panel label # 
text(0.7,2.3E6 ,labels="A", col="black", cex=2)

# P-value label
mtext(text = expression(italic("P")~" = 0.016") , side =3, line = -1.2, adj = 0.925, col="black", cex=0.8)

# Panel 2: Rpf effects on fungal abundance # 
abun.fun.fig <- plot(jitter(rep(1, length(fun.con$value)), amount = 0.1), fun.con$value, 
      ylim = c(0, 1.5E6), xlim = c(0.5, 2.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 3.5,
      cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 2,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
points(jitter(rep(2, length(fun.rpf$value)), amount = 0.1), fun.rpf$value, pch = 21, 
       bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

# Adding mean data pointfor each treatment # 
points(1, mean(fun.con$value), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(2, mean(fun.rpf$value), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  

box(lwd = 2)

# Y axis labels
mtext(expression('ITS gene copy / g soil'), side = 2,
      outer = FALSE, cex = 1, line = 3.5, adj = 0.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1, las = 1, 
     labels = c("0.0E0", "5.0E5", "1.0E6", "1.5E6"), 
     at = c(0.0E0, 5.0E5, 1.0E6, 1.5E6))

axis(side = 4, lwd.ticks = 2, cex.axis = 1, las = 1,
     at=c(0.0E0, 5.0E5, 1.0E6, 1.5E6), labels = F, tck = -0.05)

axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("Rpf-", "Rpf+"), at = c(1, 2))

axis(side = 3, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at = c(1, 2), labels = F, tck = -0.05)

# Adding SEM # 
arrows(x0 = c(1,2), y0 = fun.table$mean, y1 = fun.table$LCI, angle = 90,
       length = 0.15, lwd = 2)
arrows(x0 = c(1,2), y0 = fun.table$mean, y1 = fun.table$UCI, angle = 90,
       length=0.15, lwd = 2)

# Panel label # 
text(0.7,1.45E6 ,labels="B", col="black", cex=2)

# p-value 
mtext(text = expression(italic("P")~" = 0.007") , side =3, line = -1.2, adj = 0.925, col="black", cex=0.8)

# Panel 3: Rpf effects on fungal abundance # 
FB.fig <- plot(jitter(rep(1, length(FB.con$FB)), amount = 0.1), FB.con$FB, 
      ylim = c(0, 1.5), xlim = c(0.5, 2.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 3.5,
      cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 2,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
points(jitter(rep(2, length(FB.rpf$FB)), amount = 0.1), FB.rpf$FB, pch = 21, 
       bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

# Adding mean data pointfor each treatment # 
points(1, mean(FB.con$FB), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(2, mean(FB.rpf$FB), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  

box(lwd = 2)

# Y axis labels
mtext(expression('Fungi : bacteria ratio'), side = 2,
      outer = FALSE, cex = 1, line = 2.5, adj = 0.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1, las = 1, 
     labels = c("0.0", "0.5", "1.0", "1.5"), 
     at = c(0.0, 0.5, 1.0, 1.5))

axis(side = 4, lwd.ticks = 2, cex.axis = 1, las = 1,
     at=c(0.0, 0.5, 1.0, 1.5), labels = F, tck = -0.05)

axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("Rpf-", "Rpf+"), at = c(1, 2))

axis(side = 3, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at = c(1, 2), labels = F, tck = -0.05)

# Adding SEM # 
arrows(x0 = c(1,2), y0 = FB.table$mean, y1 = FB.table$LCI, angle = 90,
       length = 0.15, lwd = 2)
arrows(x0 = c(1,2), y0 = FB.table$mean, y1 = FB.table$UCI, angle = 90,
       length=0.15, lwd = 2)

# Panel label # 
text(0.7,1.45,labels="C", col="black", cex=2)

# p-value 
mtext(text = expression(italic("P")~" = 0.018") , side =3, line = -1.2, adj = 0.925, col="black", cex=0.8)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/Figure3-MicrobialAbundanceRatio.png")
grid.raster(img)


# Statistical tests: 2 sample t-test for Rpf effects on microbial abundance
# Bacterial abundance
bac.ttest <- t.test(stdGeneCopy ~ Treatment, data = data.sub.bac)
bac.ttest    # Significant: t = 2.7069, df = 14.777, p = 0.0164

# Fungal abundance 
fun.ttest <- t.test(stdGeneCopy ~ Treatment, data = data.sub.fun)
fun.ttest    # Significant: t = -3.2719, df = 11.701, p = 0.00689

# F:B ratio
FB.ttest <- t.test(FB ~ Treatment, data = ratioFB)
FB.ttest     # Significant: t= -2.8454, df = 9.4476, p = 0.01835
```


# 4) Effect of Rpf and metabolic status on soil bacterial community structure

## Set up work environment and load packages 

```{r, message=FALSE, warning=FALSE}
# Setup work enviroment 
rm(list = ls())
knitr::opts_knit$set(root.dir = "~/../Github/BrassicaRpf/analysis")

# Source diversity code functions from bin folder # 
source("C:/Users/vkuo/GitHub/BrassicaRpf/bin/DiversityFunctions.R")
source("C:/Users/vkuo/GitHub/BrassicaRpf/bin/MothurTools.R")
source("C:/Users/vkuo/GitHub/BrassicaRpf/bin/phylodiversity2.R")

# Load packages # 
package.list <- c('vegan', 'plyr' , 'car', 'grid', 'png', 'ape', 'picante', 'ade4', 'phytools', 'phangorn', 'indicspecies', 'viridis' ,  'BiodiversityR') 
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
  } }

# Load sem custom functions # 
sem <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}

# Load t-test custom functions # 
ttest <- function(reg, coefnum, val){
  co <- coef(summary(reg))
  tstat <- (co[coefnum,1]-val)/co[coefnum,2]
  pstat <- 2 * pt(abs(tstat), reg$df.residual, lower.tail = FALSE)
  return(list = c(t = tstat, df = reg$df.residual, p =  pstat))
}

# Run all alpha diversity analyses
run.all <- TRUE


  # design = general design file for experiment
  # shared = OTU table from mothur with sequence similarity clustering
  # tax    = Taxonomy for 97% similarity OTUs
design <- "C:/Users/vkuo/GitHub/BrassicaRpf/data/Brassica.design.txt"
shared <- "C:/Users/vkuo/GitHub/BrassicaRpf/mothur/output/Brassica.bac.final.shared"
taxon <- "C:/Users/vkuo/GitHub/BrassicaRpf/mothur/output/Brassica.bac.final.0.03.taxonomy"

# Import Design 
design <- read.delim(design, header=T, row.names=1)
# Import Shared Files
OTU <- read.otu(shared = shared, cutoff = "0.03") # 97% similarity 
# Import Taxonomy
OTU.tax <- read.tax(taxonomy = taxon, format = "rdp")  
# Import Phylogenetic tree 
OTU.tre <- read.tree("../phylo/Brassica.bac.rename.tree.2") 
```


## Soil bacteria beta and alpha diversity 


```{r}
### Sequence Coverage 
# Remove OTUs with less than two occurences across all sites # 
OTU <- OTU[, which(colSums(OTU) >= 2)]

# Remove mock community # 
OTU <- OTU[1:20, ]  

# Determine coverage of sequences # 
cov.seqs <- count.groups(OTU)
cov.mean <- mean(cov.seqs) # 160,871
cov.sem <- sem(cov.seqs) # 16,095.38
cov.min <- min(cov.seqs) # 79,797
total.seqs <- sum(cov.seqs) # 3,217,419 

# Good's coverage
goods.c <- function(x = ""){
              1 - (apply(OTU, 1, function(x){sum(x == 1)}) / rowSums(x))
}

goods.c.Brassica <- goods.c(OTU)
mean.good.c <- mean(goods.c.Brassica) # 0.984  Good mean coverage
min.good.c <- min(goods.c.Brassica) # 0.967  Good lowest coverage



### Alpha diversity 
# Resampling code to estimate alpha diversity (used if run.all = T)
if (run.all == TRUE){
  rich <- round(richness.iter(input = OTU, size = 1000,
                              iters = 100, shared = "FALSE"), 3)
  even <- round(evenness.iter(input = OTU, size = 1000,
                              iters = 100, shared = "FALSE",
                              method = "simp_even"), 3)
  rare <- rarefy(OTU, 1000, se = FALSE, MARGIN = 1)
  # Write output to files
  write.table(rich, "../data/rich.txt", sep = "\t",
              col.names = T, row.names = T)
  write.table(even, "../data/even.txt", sep = "\t",
              col.names = T, row.names = T)
}

# Read in alpha diversity files from above
rich2 <- read.table("../data/rich.txt", sep = "\t")
even2 <- read.table("../data/even.txt", sep = "\t")

# Merge data to design and calculate mean and sem per sample
rich.data <- merge(design, rich2, by = "row.names")
row.names(rich.data) <- rich.data$Row.names
rich.data <- rich.data[sort(row.names(rich.data)), ]
rich.mean <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, mean, na.rm = TRUE),3)
rich.sem <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, sem, na.rm = TRUE), 3)

even.data <- merge(design, even2, by = "row.names")
row.names(even.data) <- even.data$Row.names
even.data <- even.data[sort(row.names(even.data)), ]
even.mean <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, mean, na.rm = TRUE),3)
even.sem <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, sem, na.rm = TRUE),4)

# Make new dataframe merging design file and mean diversity
Brassica.div <- data.frame(design[sort(row.names(design)), ], rich.mean, even.mean)

# Take averages of technial reps
rich.rep.ave <- ddply(Brassica.div, .(treatment, type, rep), summarize, rich = mean(rich.mean))
even.rep.ave <- ddply(Brassica.div, .(treatment, type, rep), summarize, even = mean(even.mean))

# Reshape data 
rich.2 <- reshape(rich.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep"), direction = "wide")

even.2 <- reshape(even.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep"), direction = "wide")


## Statistical test: One-way ANOVA of Rpf and metabolic status
# Soil bacterial richness
rich.anova.c <- aov(rich.mean ~ treatment*type, Brassica.div)
summary(rich.anova.c)

# Soil bacterial evenness 
even.anova.c <- aov(even.mean ~ treatment*type, Brassica.div)
summary(even.anova.c)


### Beta Diversity
# Make presence-absence matrix
OTU.PA <- (OTU > 0) * 1

# Make relative abundence matrix
OTU.REL <- OTU
for (i in 1:dim(OTU)[1]){
  OTU.REL[i,] <- OTU[i,]/sum(OTU[i,])
  }

# Log-transform relative abundances
OTU.REL.log <- decostand(OTU, method="log")

### Just RNA ####
#OTU.REL.log <- decostand(OTU, method="log")
########

Brassica.bc.dis <- vegdist(OTU.REL.log, method = "bray", binary = "FALSE")
Brassica.dis.mean <- mean(Brassica.bc.dis)

# Principal Coordinates Analysis (PCoA)
Brassica.PCoA <- cmdscale(Brassica.bc.dis, eig = TRUE, k = 3)
explainvar1 <- round(Brassica.PCoA$eig[1] / sum(Brassica.PCoA$eig), 3) * 100
explainvar2 <- round(Brassica.PCoA$eig[2] / sum(Brassica.PCoA$eig), 3) * 100
explainvar3 <- round(Brassica.PCoA$eig[3] / sum(Brassica.PCoA$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

# OTU Scores
otu.scores <- t(cor(Brassica.PCoA$points, OTU.REL))
otu.scores <- as.matrix(otu.scores)[,1:2]
otu.scores <- otu.scores[abs(otu.scores[,1]) > 0.7|abs(otu.scores[,2]) > 0.7,]

# Average BC Distance Between Treatments
Brassica.bc.dis.m <- as.matrix(Brassica.bc.dis)
all.equal(row.names(Brassica.div), rownames(Brassica.bc.dis.m))

treatment.div <- unique(Brassica.div$treatment)
treatment.dis <- rep(NA, length(treatment.div))
for(i in 1:length(treatment.div)){
  temp <- row.names(Brassica.div[Brassica.div$treatment == treatment.div[i], ])
  treatment.dis[i] <- Brassica.bc.dis.m[temp[1], temp[2]]
}

mean(treatment.dis)

# Plot figure # 
png(filename="../figures/SupplementaryFigure-AllBacteriaOrdination.png",
    width = 1800, height = 800, res = 96*2)

layout(matrix(1:3, 1, 3), widths = c(20, 9, 2.5))

par(mar = c(7, 10, 1, 0) + 0.5)

plot(Brassica.PCoA$points[ ,1], Brassica.PCoA$points[ ,2],
     ylim = c(-0.4, 0.4), xlim = c(-0.5, 0.4),
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""), line = 5,
     #xlab = "", ylab = "", xaxt = "n", yaxt = "n",
     pch = 22, cex = 2.0, type = "n", cex.lab = 2.5, cex.axis = 3,
     axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 3, cex.axis = 2, las = 1, tck=-0.025)
axis(side = 2, labels = T, lwd.ticks = 3, cex.axis = 2, las = 1, tck=-0.025)
axis(side = 3, labels = F, lwd.ticks = 3, cex.axis = 1, las = 1, tck=-0.025)
axis(side = 4, labels = F, lwd.ticks = 3, cex.axis = 1, las = 1, tck=-0.025)
abline(h = 0, v = 0, lty = 3)

# Panel label # 
#text(-0.45, 0.35, labels="A", col="black", cex=3)

box(lwd = 2)

# Subset data
all.equal(row.names(Brassica.PCoA$points), rownames(Brassica.div))
Brassica.points <- data.frame(Brassica.PCoA$points, Brassica.div)

# Active community
Brassica.active.rpf <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.active.no <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf-"), ]
# Total community
Brassica.total.rpf <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.total.no <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf-"), ]

# Add points
# Active community Rpf+
points(Brassica.active.rpf[ ,1], Brassica.active.rpf[ ,2], pch = 21,
       cex = 3.5, col = "Black", bg = "grey15", lwd= 2.5)
# Active community Rpf-
points(Brassica.active.no[ ,1], Brassica.active.no[ ,2], pch = 21,
       cex = 3.5, col = "Black", bg = "lightgrey", lwd= 2.5)   
# Total community Rpf+ 
points(Brassica.total.rpf[ ,1], Brassica.total.rpf[ ,2], pch = 24,
       cex = 3.5, col = "Black", bg = "grey15", lwd= 2.5)
# Total community Rpf-
points(Brassica.total.no[ ,1], Brassica.total.no[ ,2], pch = 24,
       cex = 3.5, col = "Black", bg = "lightgrey", lwd= 2.5)

# Add elipcses 
# Active Rpf+
ordiellipse(cbind(Brassica.active.rpf[ ,1], Brassica.active.rpf[ ,2]), Brassica.active.rpf$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)
# Active Rpf-
ordiellipse(cbind(Brassica.active.no[ ,1], Brassica.active.no[ ,2]), Brassica.active.no$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)
# Total Rpf+
ordiellipse(cbind(Brassica.total.rpf[ ,1], Brassica.total.rpf[ ,2]), Brassica.total.rpf$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)
# Total Rpf- 
ordiellipse(cbind(Brassica.total.no[ ,1], Brassica.total.no[ ,2]), Brassica.total.no$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)

# Add Legend Outside
par(mar = c(4, 0, 5, 1) + 0.5)
plot.new()
legend(0, 1, c("Active Rpf-", "Active Rpf+", "Total Rpf-", "Total Rpf+"),
       pch = c(21, 21, 24, 24),
       col = c("Black", "Black", "Black", "Black"),
       pt.bg = c("lightgrey", "grey15", "lightgrey", "grey15"),
       bty = "n", y.intersp = 1, pt.cex = 3.2, cex = 2, lwd= 2, lty =  NA)


# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/SupplementaryFigure-AllBacteriaOrdination.png")
grid.raster(img)

# Add column for pot number to account of paired/match design
Brassica.div$pot <- c(1,2,3,4,5,6,7,8,9,10,1,2,3,4,5,6,7,8,9,10)


## Statistical test: PERMANOVA to test Rpf and metabolic status effects on soil bacterial community structure
bray.permanova <- adonis(OTU.REL.log ~ Brassica.div$type + Brassica.div$treatment + Brassica.div$pot + 
                           Brassica.div$type*Brassica.div$treatment,
                         method = "bray", binary = FALSE, permutations = 999)
bray.permanova    
    # Metabolic status: Significant:                 r2 = 0.22571, p = 0.001 ***
    # Rpf:              Marginally non-significant:  r2 = 0.06086, p = 0.083 . 
    # Interaction:      Non-significant:             r2 = 0.03988, p = 0.447 

#bray.rna.permanova <- adonis(OTU.REL.log ~ Brassica.div[11:20, ]$treatment, method = "bray", binary = FALSE, permutations=999)
#bray.rna.permanova
```


## Gram positive bacteria alpha and beta- diversity 


```{r}
# Subset Gram positive OTU from taxonomy file
GramPositive.OTU.tax <- OTU.tax[ which(OTU.tax$Phylum == 'Actinobacteria' | OTU.tax$Phylum == 'Firmicutes'), ]

# Subset OTU table to contain only Gram positive bacteria
GramPositive.OTU <- match(GramPositive.OTU.tax$OTU, colnames(OTU))
GramPositive.OTU <- sort(c(GramPositive.OTU-1, GramPositive.OTU))
Gpos.OTU <- OTU[, GramPositive.OTU]

# Remove OTUs with less than two occurences across all sites # 
Gpos.OTU <- Gpos.OTU[, which(colSums(Gpos.OTU) >= 2)]

# Remove mock community # 
Gpos.OTU <- Gpos.OTU[1:20, ]


### Alpha diversity 
# Resampling code to estimate alpha diversity (used if run.all = T)
if (run.all == TRUE){
  rich <- round(richness.iter(input = Gpos.OTU, size = 1000,
                              iters = 100, shared = "FALSE"), 3)
  even <- round(evenness.iter(input = Gpos.OTU, size = 1000,
                              iters = 100, shared = "FALSE",
                              method = "simp_even"), 3)
  rare <- rarefy(Gpos.OTU, 1000, se = FALSE, MARGIN = 1)
  # Write output to files
  write.table(rich, "../data/rich.txt", sep = "\t",
              col.names = T, row.names = T)
  write.table(even, "../data/even.txt", sep = "\t",
              col.names = T, row.names = T)
}

# Read in alpha diversity files from above
rich2 <- read.table("../data/rich.txt", sep = "\t")
even2 <- read.table("../data/even.txt", sep = "\t")

# Merge data to design and calculate mean and sem per sample
rich.data <- merge(design, rich2, by = "row.names")
row.names(rich.data) <- rich.data$Row.names
rich.data <- rich.data[sort(row.names(rich.data)), ]
rich.mean <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, mean, na.rm = TRUE),3)
rich.sem <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, sem, na.rm = TRUE), 3)

even.data <- merge(design, even2, by = "row.names")
row.names(even.data) <- even.data$Row.names
even.data <- even.data[sort(row.names(even.data)), ]
even.mean <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, mean, na.rm = TRUE),3)
even.sem <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, sem, na.rm = TRUE),4)

# Make new dataframe merging design file and mean diversity
Brassica.div <- data.frame(design[sort(row.names(design)), ], rich.mean, even.mean)

# Take averages of technial reps
rich.rep.ave <- ddply(Brassica.div, .(treatment, type, rep), summarize, rich = mean(rich.mean))
even.rep.ave <- ddply(Brassica.div, .(treatment, type, rep), summarize, even = mean(even.mean))

# Reshape data 
rich.2 <- reshape(rich.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep"), direction = "wide")

even.2 <- reshape(even.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep"), direction = "wide")


## Statistical test: two-way ANOVA of Rpf and metabolic status
# Soil bacterial richness
rich.anova.c <- aov(rich.mean ~ type + treatment + treatment*type, Brassica.div)
summary(rich.anova.c)

# Soil bacterial evenness 
even.anova.c <- aov(even.mean ~ type + treatment + treatment*type, Brassica.div)
summary(even.anova.c)


### Beta Diversity
# Make presence-absence matrix
Gpos.OTU.PA <- (Gpos.OTU > 0) * 1

# Make relative abundence matrix
Gpos.OTU.REL <- Gpos.OTU
for (i in 1:dim(Gpos.OTU)[1]){
  Gpos.OTU.REL[i,] <- Gpos.OTU[i,]/sum(Gpos.OTU[i,])
  }

# Log-transform relative abundances
Gpos.OTU.REL.log <- decostand(Gpos.OTU, method="log")

Brassica.bc.dis <- vegdist(Gpos.OTU.REL.log, method = "bray", binary = "FALSE")
Brassica.dis.mean <- mean(Brassica.bc.dis)

# Principal Coordinates Analysis (PCoA)
Brassica.PCoA <- cmdscale(Brassica.bc.dis, eig = TRUE, k = 3)
explainvar1 <- round(Brassica.PCoA$eig[1] / sum(Brassica.PCoA$eig), 3) * 100
explainvar2 <- round(Brassica.PCoA$eig[2] / sum(Brassica.PCoA$eig), 3) * 100
explainvar3 <- round(Brassica.PCoA$eig[3] / sum(Brassica.PCoA$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

# Gpos.OTU Scores
Gpos.OTU.scores <- t(cor(Brassica.PCoA$points, Gpos.OTU.REL))
Gpos.OTU.scores <- as.matrix(Gpos.OTU.scores)[,1:2]
Gpos.OTU.scores <- Gpos.OTU.scores[abs(Gpos.OTU.scores[,1]) > 0.7|abs(Gpos.OTU.scores[,2]) > 0.7,]

# Average BC Distance Between Treatments
Brassica.bc.dis.m <- as.matrix(Brassica.bc.dis)
all.equal(row.names(Brassica.div), rownames(Brassica.bc.dis.m))

treatment.div <- unique(Brassica.div$treatment)
treatment.dis <- rep(NA, length(treatment.div))
for(i in 1:length(treatment.div)){
  temp <- row.names(Brassica.div[Brassica.div$treatment == treatment.div[i], ])
  treatment.dis[i] <- Brassica.bc.dis.m[temp[1], temp[2]]
}

mean(treatment.dis)


# Plot figure # 
png(filename="../figures/Figure5-GramPositiveOrdination.png",
    width = 1800, height = 800, res = 96*2)

layout(matrix(1:3, 1, 3), widths = c(20, 9, 2.5))

par(mar = c(7, 10, 1, 0) + 0.5)

plot(Brassica.PCoA$points[ ,1], Brassica.PCoA$points[ ,2],
     ylim = c(-0.4, 0.4), xlim = c(-0.5, 0.4),
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""), line = 5,
     #xlab = "", ylab = "", xaxt = "n", yaxt = "n",
     pch = 22, cex = 2.0, type = "n", cex.lab = 2.5, cex.axis = 3,
     axes = FALSE)


# Add Axes
axis(side = 1, labels = T, lwd.ticks = 3, cex.axis = 2, las = 1, tck=-0.025)
axis(side = 2, labels = T, lwd.ticks = 3, cex.axis = 2, las = 1, tck=-0.025)
axis(side = 3, labels = F, lwd.ticks = 3, cex.axis = 1, las = 1, tck=-0.025)
axis(side = 4, labels = F, lwd.ticks = 3, cex.axis = 1, las = 1, tck=-0.025)
abline(h = 0, v = 0, lty = 3)


# Panel label # 
#text(-0.45, 0.35, labels="B", col="black", cex=3)

box(lwd = 2)

# Subset data
all.equal(row.names(Brassica.PCoA$points), rownames(Brassica.div))
Brassica.points <- data.frame(Brassica.PCoA$points, Brassica.div)

# Active community
Brassica.active.rpf <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.active.no <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf-"), ]
# Total community
Brassica.total.rpf <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.total.no <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf-"), ]

# Add points
# Active community Rpf+
points(Brassica.active.rpf[ ,1], Brassica.active.rpf[ ,2], pch = 21,
       cex = 3.5, col = "Black", bg = "grey15", lwd= 2.5)
# Active community Rpf-
points(Brassica.active.no[ ,1], Brassica.active.no[ ,2], pch = 21,
       cex = 3.5, col = "Black", bg = "lightgrey", lwd= 2.5)   
# Total community Rpf+ 
points(Brassica.total.rpf[ ,1], Brassica.total.rpf[ ,2], pch = 24,
       cex = 3.5, col = "Black", bg = "grey15", lwd= 2.5)
# Total community Rpf-
points(Brassica.total.no[ ,1], Brassica.total.no[ ,2], pch = 24,
       cex = 3.5, col = "Black", bg = "lightgrey", lwd= 2.5)

# Add elipcses 
# Active Rpf+
ordiellipse(cbind(Brassica.active.rpf[ ,1], Brassica.active.rpf[ ,2]), Brassica.active.rpf$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)
# Active Rpf-
ordiellipse(cbind(Brassica.active.no[ ,1], Brassica.active.no[ ,2]), Brassica.active.no$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)
# Total Rpf+
ordiellipse(cbind(Brassica.total.rpf[ ,1], Brassica.total.rpf[ ,2]), Brassica.total.rpf$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)
# Total Rpf- 
ordiellipse(cbind(Brassica.total.no[ ,1], Brassica.total.no[ ,2]), Brassica.total.no$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)


# Add Legend Outside
par(mar = c(4, 0, 5, 1) + 0.5)
plot.new()
legend(0, 1, c("Active Rpf-", "Active Rpf+", "Total Rpf-", "Total Rpf+"),
       pch = c(21, 21, 24, 24),
       col = c("Black", "Black", "Black", "Black"),
       pt.bg = c("lightgrey", "grey15", "lightgrey", "grey15"),
       bty = "n", y.intersp = 1, pt.cex = 3.2, cex = 2, lwd= 2, lty =  NA)


# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/Figure5-GramPositiveOrdination.png")
grid.raster(img)


Brassica.div$pot <- c(1,2,3,4,5,6,7,8,9,10,1,2,3,4,5,6,7,8,9,10)
Brassica.div$pot <- as.factor(Brassica.div$pot)

## Statistical test: PERMANOVA to test Rpf and metabolic status effects on soil bacterial community structure
bray.permanova <- adonis(Gpos.OTU.REL ~ Brassica.div$type + Brassica.div$treatment + Brassica.div$pot + 
                           Brassica.div$type*Brassica.div$treatment,
                         method = "bray", binary = FALSE, permutations = 999)
bray.permanova    
    # Metabolic status: Significant:                 r2 = 0.20609, p = 0.001 ***
    # Rpf:              Marginally non-significant:  r2 = 0.06795, p = 0.053 . 
    # Interaction:      Non-significant:             r2 = 0.03812, p = 0.603
```


## Actinobacteria alpha and beta- diversity 


```{r}
# Subset Gram positive OTU from taxonomy file
Actino.OTU.tax <- OTU.tax[ which(OTU.tax$Phylum == 'Actinobacteria'), ]

# Subset OTU table to contain only Gram positive bacteria
Actino.OTU <- match(Actino.OTU.tax$OTU, colnames(OTU))
Actino.OTU <- sort(c(Actino.OTU-1, Actino.OTU))
Actino.OTU <- OTU[, Actino.OTU]

# Remove OTUs with less than two occurences across all sites # 
Actino.OTU <- Actino.OTU[, which(colSums(Actino.OTU) >= 2)]

# Remove mock community # 
Actino.OTU <- Actino.OTU[1:20, ]


### Alpha diversity 
# Resampling code to estimate alpha diversity (used if run.all = T)
if (run.all == TRUE){
  rich <- round(richness.iter(input = Actino.OTU, size = 1000,
                              iters = 100, shared = "FALSE"), 3)
  even <- round(evenness.iter(input = Actino.OTU, size = 1000,
                              iters = 100, shared = "FALSE",
                              method = "simp_even"), 3)
  rare <- rarefy(Actino.OTU, 1000, se = FALSE, MARGIN = 1)
  # Write output to files
  write.table(rich, "../data/rich.txt", sep = "\t",
              col.names = T, row.names = T)
  write.table(even, "../data/even.txt", sep = "\t",
              col.names = T, row.names = T)
}

# Read in alpha diversity files from above
rich2 <- read.table("../data/rich.txt", sep = "\t")
even2 <- read.table("../data/even.txt", sep = "\t")

# Merge data to design and calculate mean and sem per sample
rich.data <- merge(design, rich2, by = "row.names")
row.names(rich.data) <- rich.data$Row.names
rich.data <- rich.data[sort(row.names(rich.data)), ]
rich.mean <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, mean, na.rm = TRUE),3)
rich.sem <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, sem, na.rm = TRUE), 3)

even.data <- merge(design, even2, by = "row.names")
row.names(even.data) <- even.data$Row.names
even.data <- even.data[sort(row.names(even.data)), ]
even.mean <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, mean, na.rm = TRUE),3)
even.sem <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, sem, na.rm = TRUE),4)

# Make new dataframe merging design file and mean diversity
Brassica.div <- data.frame(design[sort(row.names(design)), ], rich.mean, even.mean)

# Take averages of technial reps
rich.rep.ave <- ddply(Brassica.div, .(treatment, type, rep), summarize, rich = mean(rich.mean))
even.rep.ave <- ddply(Brassica.div, .(treatment, type, rep), summarize, even = mean(even.mean))

# Reshape data 
rich.2 <- reshape(rich.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep"), direction = "wide")

even.2 <- reshape(even.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep"), direction = "wide")


## Statistical test: two-way ANOVA of Rpf and metabolic status
# Soil bacterial richness
rich.anova.c <- aov(rich.mean ~ type + treatment + treatment*type, Brassica.div)
summary(rich.anova.c)

# Soil bacterial evenness 
even.anova.c <- aov(even.mean ~ type + treatment + treatment*type, Brassica.div)
summary(even.anova.c)


### Beta Diversity
# Make presence-absence matrix
Actino.OTU.PA <- (Actino.OTU > 0) * 1

# Make relative abundence matrix
Actino.OTU.REL <- Actino.OTU
for (i in 1:dim(Actino.OTU)[1]){
  Actino.OTU.REL[i,] <- Actino.OTU[i,]/sum(Actino.OTU[i,])
  }

# Log-transform relative abundances
Actino.OTU.REL.log <- decostand(Actino.OTU, method="log")

Brassica.bc.dis <- vegdist(Actino.OTU.REL.log, method = "bray", binary = "FALSE")
Brassica.dis.mean <- mean(Brassica.bc.dis)

# Principal Coordinates Analysis (PCoA)
Brassica.PCoA <- cmdscale(Brassica.bc.dis, eig = TRUE, k = 3)
explainvar1 <- round(Brassica.PCoA$eig[1] / sum(Brassica.PCoA$eig), 3) * 100
explainvar2 <- round(Brassica.PCoA$eig[2] / sum(Brassica.PCoA$eig), 3) * 100
explainvar3 <- round(Brassica.PCoA$eig[3] / sum(Brassica.PCoA$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

# Actino.OTU Scores
Actino.OTU.scores <- t(cor(Brassica.PCoA$points, Actino.OTU.REL))
Actino.OTU.scores <- as.matrix(Actino.OTU.scores)[,1:2]
Actino.OTU.scores <- Actino.OTU.scores[abs(Actino.OTU.scores[,1]) > 0.7|abs(Actino.OTU.scores[,2]) > 0.7,]

# Average BC Distance Between Treatments
Brassica.bc.dis.m <- as.matrix(Brassica.bc.dis)
all.equal(row.names(Brassica.div), rownames(Brassica.bc.dis.m))

treatment.div <- unique(Brassica.div$treatment)
treatment.dis <- rep(NA, length(treatment.div))
for(i in 1:length(treatment.div)){
  temp <- row.names(Brassica.div[Brassica.div$treatment == treatment.div[i], ])
  treatment.dis[i] <- Brassica.bc.dis.m[temp[1], temp[2]]
}

mean(treatment.dis)


# Plot figure # 
png(filename="../figures/Figure5-ActinoOrdination.png",
    width = 1800, height = 800, res = 96*2)

layout(matrix(1:3, 1, 3), widths = c(20, 9, 2.5))

par(mar = c(7, 10, 1, 0) + 0.5)

plot(Brassica.PCoA$points[ ,1], Brassica.PCoA$points[ ,2],
     ylim = c(-0.4, 0.4), xlim = c(-0.5, 0.4),
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""), line = 5,
     #xlab = "", ylab = "", xaxt = "n", yaxt = "n",
     pch = 22, cex = 2.0, type = "n", cex.lab = 2.5, cex.axis = 3,
     axes = FALSE)


# Add Axes
axis(side = 1, labels = T, lwd.ticks = 3, cex.axis = 2, las = 1, tck=-0.025)
axis(side = 2, labels = T, lwd.ticks = 3, cex.axis = 2, las = 1, tck=-0.025)
axis(side = 3, labels = F, lwd.ticks = 3, cex.axis = 1, las = 1, tck=-0.025)
axis(side = 4, labels = F, lwd.ticks = 3, cex.axis = 1, las = 1, tck=-0.025)
abline(h = 0, v = 0, lty = 3)


# Panel label # 
#text(-0.45, 0.35, labels="B", col="black", cex=3)

box(lwd = 2)

# Subset data
all.equal(row.names(Brassica.PCoA$points), rownames(Brassica.div))
Brassica.points <- data.frame(Brassica.PCoA$points, Brassica.div)

# Active community
Brassica.active.rpf <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.active.no <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf-"), ]
# Total community
Brassica.total.rpf <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.total.no <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf-"), ]

# Add points
# Active community Rpf+
points(Brassica.active.rpf[ ,1], Brassica.active.rpf[ ,2], pch = 21,
       cex = 3.5, col = "Black", bg = "grey15", lwd= 2.5)
# Active community Rpf-
points(Brassica.active.no[ ,1], Brassica.active.no[ ,2], pch = 21,
       cex = 3.5, col = "Black", bg = "lightgrey", lwd= 2.5)   
# Total community Rpf+ 
points(Brassica.total.rpf[ ,1], Brassica.total.rpf[ ,2], pch = 24,
       cex = 3.5, col = "Black", bg = "grey15", lwd= 2.5)
# Total community Rpf-
points(Brassica.total.no[ ,1], Brassica.total.no[ ,2], pch = 24,
       cex = 3.5, col = "Black", bg = "lightgrey", lwd= 2.5)

# Add elipcses 
# Active Rpf+
ordiellipse(cbind(Brassica.active.rpf[ ,1], Brassica.active.rpf[ ,2]), Brassica.active.rpf$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)
# Active Rpf-
ordiellipse(cbind(Brassica.active.no[ ,1], Brassica.active.no[ ,2]), Brassica.active.no$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)
# Total Rpf+
ordiellipse(cbind(Brassica.total.rpf[ ,1], Brassica.total.rpf[ ,2]), Brassica.total.rpf$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)
# Total Rpf- 
ordiellipse(cbind(Brassica.total.no[ ,1], Brassica.total.no[ ,2]), Brassica.total.no$treatment, kind="sd", conf=0.95,
            lwd=2, lty=3, draw="lines", col="black", label=FALSE)


# Add Legend Outside
par(mar = c(4, 0, 5, 1) + 0.5)
plot.new()
legend(0, 1, c("Active Rpf-", "Active Rpf+", "Total Rpf-", "Total Rpf+"),
       pch = c(21, 21, 24, 24),
       col = c("Black", "Black", "Black", "Black"),
       pt.bg = c("lightgrey", "grey15", "lightgrey", "grey15"),
       bty = "n", y.intersp = 1, pt.cex = 3.2, cex = 2, lwd= 2, lty =  NA)


# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/Figure5-ActinoOrdination.png")
grid.raster(img)


#Brassica.div$pot <- c(1,2,3,4,5,6,7,8,9,10,1,2,3,4,5,6,7,8,9,10)
#Brassica.div$pot <- as.factor(Brassica.div$pot)

## Statistical test: PERMANOVA to test Rpf and metabolic status effects on soil bacterial community structure
#bray.permanova <- adonis(Actino.OTU.REL ~ Brassica.div$type + Brassica.div$treatment + Brassica.div$pot + 
#                           Brassica.div$type*Brassica.div$treatment,
#                         method = "bray", binary = FALSE, permutations = 999)
bray.permanova <- adonis(Actino.OTU.REL ~ Brassica.div$type + Brassica.div$treatment + 
                           Brassica.div$type*Brassica.div$treatment,
                         method = "bray", binary = FALSE, permutations = 999)
bray.permanova    
    # Metabolic status: Significant:                 r2 = 0.33, p = 0.001 ***
    # Rpf:              Marginally non-significant:  r2 = 0.067, p = 0.053 . 
    # Interaction:      Non-significant:             r2 = 0.04, p = 0.276
```





# 6) Supplementary materials

## Gram positive: Gram negative ratios

```{r}
# Subsetting 
OTU.Rpf <- OTU[c(6:10, 16:20), ]
Gpos.OTU.Rpf <- Gpos.OTU[c(6:10, 16:20), ]

OTU.Con <- OTU[c(1:5, 11:15), ]
Gpos.OTU.Con <- Gpos.OTU[c(1:5, 11:15), ]
  
# Calculate sum reads for gram positive 
Gram.pos.Rpf <- as.vector(rowSums(Gpos.OTU.Rpf))      
All.Rpf <- as.vector(rowSums(OTU.Rpf))  
Gram.pos.Con <- as.vector(rowSums(Gpos.OTU.Con))       
All.Con <- as.vector(rowSums(OTU.Con))  

# Calculate Gram negative read sum 
Gram.neg.Rpf <- All.Rpf - Gram.pos.Rpf  
Gram.neg.Con <- All.Con - Gram.pos.Con  

# Calculate ratio of Gram postive : Gram negative 
Rpf.Gram.Ratio <- Gram.pos.Rpf / Gram.neg.Rpf
Con.Gram.Ratio <- Gram.pos.Con / Gram.neg.Con

Gram.Ratio.ttest <- t.test(Con.Gram.Ratio, Rpf.Gram.Ratio, alternative="greater")
Gram.Ratio.ttest  # Non-significant: t = 1.2977, df = 15.203, p = 0.1069

mean.Rpf.Gram.Ratio <- mean(Rpf.Gram.Ratio)
mean.Con.Gram.Ratio <- mean(Con.Gram.Ratio)

df <- rbind(as.numeric(mean.Rpf.Gram.Ratio), as.numeric(mean.Con.Gram.Ratio))
df <- as.data.frame(df)
df$Treatment <- c("Rpf+", "Rpf-")
colnames(df) <- c("Ratio", "Treatment")

barplot(df$Ratio, names.arg=c("Rpf+", "Rpf-") )

```


## Actinobacteria: Gram negative ratios

```{r}
# Subsetting 
OTU.Rpf <- OTU[c(6:10, 16:20), ]
Actino.OTU.Rpf <- Actino.OTU[c(6:10, 16:20), ]

OTU.Con <- OTU[c(1:5, 11:15), ]
Actino.OTU.Con <- Actino.OTU[c(1:5, 11:15), ]
  
# Calculate sum reads for gram positive 
Actino.Rpf <- as.vector(rowSums(Actino.OTU.Rpf))      
All.Rpf <- as.vector(rowSums(OTU.Rpf))  
Actino.Con <- as.vector(rowSums(Actino.OTU.Con))       
All.Con <- as.vector(rowSums(OTU.Con))  

# Calculate Gram negative read sum 
Gram.neg.Rpf <- All.Rpf - Actino.Rpf  
Gram.neg.Con <- All.Con - Actino.Con  

# Calculate ratio of Gram postive : Gram negative 
Rpf.Gram.Ratio <- Actino.Rpf / Gram.neg.Rpf
Con.Gram.Ratio <- Actino.Con / Gram.neg.Con

Gram.Ratio.ttest <- t.test(Con.Gram.Ratio, Rpf.Gram.Ratio, alternative="greater")
Gram.Ratio.ttest  # Non-significant: t = 1.2977, df = 15.203, p = 0.1069

mean.Rpf.Gram.Ratio <- mean(Rpf.Gram.Ratio)
mean.Con.Gram.Ratio <- mean(Con.Gram.Ratio)

df <- rbind(as.numeric(mean.Rpf.Gram.Ratio), as.numeric(mean.Con.Gram.Ratio))
df <- as.data.frame(df)
df$Treatment <- c("Rpf+", "Rpf-")
colnames(df) <- c("Ratio", "Treatment")

barplot(df$Ratio, names.arg=c("Rpf+", "Rpf-") )

```





## Control Arabidopsis experiment 

```{r}
# Load dataset 
seedling <- read.delim("~/../Github/BrassicaRpf/data/seedlingbiomass.txt", sep = ",", head = TRUE)

# Calculate relative biomass # 
seedling$Biomass <- (seedling$BiomassPixel)/(seedling$PlatePixel)
seedling$RelativeBiomass <- (seedling$Biomass)/(seedling$Seedlings)*100

# Biomass data points # 
seedling.rpf <- seedling[ which(seedling$Treatment == "Rpf+"),]
seedling.con <- seedling[ which(seedling$Treatment == "Rpf-"),]

# Biomass data table # 
seedling.mean <- aggregate(seedling$RelativeBiomass ~ Treatment, seedling, mean)
seedling.sem <- aggregate(seedling$RelativeBiomass ~ Treatment, seedling, sem)
seedling.sem.LL <- seedling.mean[2] + seedling.sem[2]
seedling.sem.UL <- seedling.mean[2] - seedling.sem[2]
seedling.table <- data.frame(seedling.mean[1], seedling.mean[2], seedling.sem[2],
          seedling.sem.LL[1], seedling.sem.UL[1])
colnames(seedling.table) <- c("Treatment", "mean", "sem", "LCI", "UCI")
seedling.table <- seedling.table[order(seedling.table$mean),]

# Plotting Arabidopsis biomass # 
png(filename="../figures/SupplementaryFig1-Arabidopsiscontrol.png",
    width = 800, height = 800, res = 96*2)
par(mar = c(5, 5, 1, 1))

arabid.fig <- plot(jitter(rep(1, length(seedling.con$RelativeBiomass)), amount = 0.1), seedling.con$RelativeBiomass,
      ylim = c(0, 1), xlim = c(0.5, 2.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 3.5,
      cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 2,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
points(jitter(rep(2, length(seedling.rpf$RelativeBiomass)), amount = 0.1), seedling.rpf$RelativeBiomass, pch = 21, 
       bg = "gray15", col = "gray15", lwd = 2, cex = 1.7)

# Adding mean data pointfor each treatment # 
points(1, mean(seedling.con$RelativeBiomass), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(2, mean(seedling.rpf$RelativeBiomass), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  

box(lwd = 2)

# Y axis labels
mtext(expression('Relative Biomass'), side = 2,
      outer = FALSE, cex = 1.5, line = 3.5, adj = 0.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1, las = 1, 
     labels = c("0.0", "0.5", "1.0"), 
     at = c(0.0, 0.5, 1.0))

axis(side = 4, lwd.ticks = 2, cex.axis = 1, las = 1,
     at=c(0.0, 0.5, 1.0), labels = F, tck = -0.02)

axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("Rpf-", "Rpf+"), at = c(1, 2))

axis(side = 3, lwd.ticks = 2, cex.axis = 1.5, las = 1,
     at = c(1, 2), labels = F, tck = -0.02)

# Adding SEM # 
arrows(x0 = c(2,1), y0 = seedling.table$mean, y1 = seedling.table$LCI, angle = 90,
       length = 0.25, lwd = 2)
arrows(x0 = c(2,1), y0 = seedling.table$mean, y1 = seedling.table$UCI, angle = 90,
       length=0.25, lwd = 2)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/SupplementaryFig1-Arabidopsiscontrol.png")
grid.raster(img)


# Statistical test: t-test of Rpf effects on Arabdiopsis plant biomass
anova <- aov(seedling$RelativeBiomass ~ seedling$Treatment, data = seedling)
summary(anova) 
TukeyHSD(anova)  # Non-significant: p = 0.321 
```







