---
title: "Brassica manuscript figures"
author: "Venus Kuo"
date: "Janurary 14, 2018"
output: html_document
---

# 1) Question: How does resuscitating the microbial seed bank with Rpf alter plant fitness and producitivty?

Methods: Here, I performed a full factoral experiment treating Brassica rapa plants grown in growth chambers with recombiant Rpf protein in live and sterilized soil. At the end of the six week growth chamber experiment, biomass was determined by dry weight. Flower count was taken prior to harvesting and seeds were counted to determine reproductive fitness. 

# 2) Set working environment and load packages

```{r, message=FALSE, warning=FALSE}
# Set working directory #
rm(list = ls())
getwd()
setwd("~/../Github/BrassicaRpf/data")

# Require or install packages #
package.list <- c('vegan', 'nlme' ,'data.table', 'plyr', 'reshape', 'reshape2', 'ggplot2') 
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
  } }
```

# 3) Load and describe data set

Plant Fitness.csv : Plant attributes collected at end of 6 week growing period. Information include ID number, Treatment (rpf+ = 1, rpf- = 2), soil type (complex = 2, simple = 1), flower.count, seed.count, SLA (Specific leaf area), height, abiomass (above ground biomass), bbiomass (below ground biomass), dweight (dry weight), wweight (wet weight), shoot.root (shoot:root ratio), total.biomass.

```{r, message=FALSE, warning=FALSE}
setwd("~/../Github/BrassicaRpf/data")

# Load csv files # 
data <- read.csv("Plant Fitness.csv", header=T)

# Remove compromised replicate #
data <- data[-33, ]
```


# 4) Brassica rapa plant reproductive fitness 

Bargraph of plant reproductive fitness as a metric of seed and flower counts.

 - Two-factor ANOVA testing how soil type (sterlized versus live) and Rpf treatment affected seed and flower production. 

 - One-factor ANOVA testing how Rpf treatment affecte seed and flower production in live soil. 
 
 - Student t-test to test significance of Rpf treatment on seed and flower production in live soil. 

```{r}
# Averaging the seed counts for each treatment and soil type group # 
seed.mean <- ddply(data, c("Treatment", "soil"), summarise, mean=mean(seed.count))

# Barplot of mean seed counts #
seed.mean.barplot <- ggplot(seed.mean, aes(x=soil, y = mean, fill = Treatment)) + 
  geom_bar(width=0.5, colour = "black", position = position_dodge(width = 0.65), stat = "identity", size=1)

# Calculating standard error #
seed.mean.sem <- ddply(data, c("Treatment", "soil"), summarise, 
                   mean=mean(seed.count), sem=sd(seed.count)/sqrt(length(seed.count)))
seed.mean.sem <- transform(seed.mean.sem, lower=mean-sem, upper=mean+sem) 

# Plotting bar graph with standard error #
seed.mean.barplot + geom_errorbar(aes(ymax=upper, ymin=lower),
                              position = position_dodge(width = 0.65), data=seed.mean.sem, width = 0.2, size = 1) +
  labs(x = "Soil Type", y="Seeds Produced/Plant") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Black")) +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =20),
        axis.title=element_text(size=22),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(colour="black",  margin = margin(15,10,0,10)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18)) +
  annotate("text", x =0.575, y=87, label = c("A"), size=12)

```
  
  
```{r}
# Flower Production # 
# Averaging the flower counts for each treatment and soil type group # 
flower.mean <- ddply(data, c("Treatment", "soil"), summarise, mean=mean(flower.count))

# Barplot of flower production by soil type and treatment
flower.mean.barplot <- ggplot(flower.mean, aes(x=soil, y = mean, fill = Treatment)) + 
  geom_bar(width=0.5, colour = "black", position = position_dodge(width = 0.65), stat = "identity", size=1)

# Calculating standard error of the mean for flower production # 
flower.mean.sem <- ddply(data, c("Treatment", "soil"), summarise,
                   mean=mean(flower.count), sem=sd(flower.count)/sqrt(length(flower.count)))
flower.mean.sem <- transform(flower.mean.sem, lower=mean-sem, upper=mean+sem)

# Plotting bar graph with standard error #
flower.mean.barplot + geom_errorbar(aes(ymax=upper, ymin=lower),
                              position=position_dodge(.65), data=flower.mean.sem, width = 0.2, size=1) +
  labs(x = "Soil Type" , y="Flowers Produced/Plant") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Black")) +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =20),
        axis.title=element_text(size=22),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(colour="black",  margin = margin(15,10,0,10)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18)) +
 annotate("text", x =0.575, y=12.5, label = c("B"), size=12)

```

# 5) The effect of Rpf treatment and soil type on Brassica reproductive fitness?

```{r}
data.live <- subset(data, soil == "Live")

# Seed count # 
# Two-factored ANOVA of the effect of treatment and soil type on seed count # 
seed.aov <- anova(lm(seed.count ~ Treatment*soil, data=data))
seed.aov

# One-factored ANOVA testing the effect of treatment on seed counts of plants in live soil # 
seed.live.aov <- aov(seed.count ~ Treatment, data=data.live)
summary(seed.live.aov)
TukeyHSD(seed.live.aov)

# Flower count # 
# Two-factor ANOVA of flower counts with treatment and soil type # 
flower.aov <- anova(lm(flower.count ~ Treatment*soil, data=data))
flower.aov

# One-factored ANOVA testing the effect of treatment on flower counts of plants in live soil # 
flower.live.aov <- aov(flower.count ~ Treatment, data=data.live)
summary(flower.live.aov)
TukeyHSD(flower.live.aov)
# Results: Rpf treatment marginally affected reproductivie fitness. 
```

# 6) Brassica rapa total plant biomass 

Bargraph of plant total biomass

 - Two-factor ANOVA testing how soil type (sterlized versus live) and Rpf treatment affected total plant biomass. 
 
 - One-factor ANOVA testing how Rpf treatment affecte total biomass in live soil. 
 
 - Student t-test to test significance of Rpf treatment on total biomass in live soil. 

```{r}
# Summarize total plant biomass #
plantmeans <- ddply(data, c("Treatment", "soil"), summarise, mean=mean(total.biomass))

# Barplot of total plant biomass #
plantmeans.barplot<- ggplot(plantmeans, aes(x= soil , y = mean, fill = Treatment)) + 
   geom_bar(width=0.5, colour="black", position = position_dodge(width = 0.65), stat = "identity", size=1)

# Calculating standard error of means for total plant biomass #
plantmeans.sem <- ddply(data, c("soil", "Treatment"), summarise,
                   mean=mean(total.biomass), sem=sd(total.biomass)/sqrt(length(total.biomass)))
plantmeans.sem <- transform(plantmeans.sem, lower=mean-sem, upper=mean+sem)

# Barplot of total plant biomass with sem added # 
plantmeans.barplot + geom_errorbar(aes(ymax=upper, ymin=lower),
                              position=position_dodge(0.65), data=plantmeans.sem, width = 0.2, size=1) +
  labs(x = "Soil Type", y="Total Plant Biomass (g)") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Black")) +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =20),
        axis.title=element_text(size=22),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(15,1,0,10)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18)) +
  annotate("text", x =0.575, y=2.055, label = c("A"), size=12)

```


# Brassica rapa above and belowground biomass with Rpf treatment in live soil

```{r}
# Root shoot ratio calculations # 
abio <- data$abiomass
bbio <- data$bbiomass
data$root.shoot.ratio <- abio/bbio

root.shoot.means.sem <- ddply(data, c("soil", "Treatment"), summarise,
                   mean=mean(root.shoot.ratio), sem=sd(root.shoot.ratio)/sqrt(length(root.shoot.ratio)))
root.shoot.means.sem <- transform(root.shoot.means.sem, lower=mean-sem, upper=mean+sem)


bbiomass.means.sem <- ddply(data, c("soil", "Treatment"), summarise,
                   mean=mean(bbiomass), sem=sd(bbiomass)/sqrt(length(bbiomass)))
bbiomass.means.sem <- transform(bbiomass.means.sem, lower=mean-sem, upper=mean+sem)

abiomass.means.sem <- ddply(data, c("soil", "Treatment"), summarise,
                   mean=mean(abiomass), sem=sd(abiomass)/sqrt(length(abiomass)))
abiomass.means.sem <- transform(abiomass.means.sem, lower=mean-sem, upper=mean+sem)

# Bar plot of data # 
root.shoot.bar <- ggplot(root.shoot.means.sem, aes(x = soil, y = mean, fill = Treatment)) + 
  geom_bar(width=0.5, colour="black", position = position_dodge(width = 0.65), stat = "identity", size=1)
root.shoot.bar # Root:shoot decreased in live soil relative to sterilized soil and 

root.shoot.bar  + geom_errorbar(aes(ymax=upper, ymin=lower),
                              position=position_dodge(0.65), data=root.shoot.means.sem, width = 0.2, size=1) +
  labs(x =  "Soil Type" ,y="Plant Root:Shoot Ratio") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Black")) +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =20),
        axis.title=element_text(size=22),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(15,10,0,10)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18)) #+
#  annotate("text", x =0.55, y=1.1, label = c("B"), size=12)

```



```{r}
# Belowground biomass # 
# Subsetting live soil data # 
plant.sub.ab <- subset(data, soil=="Live", select=(c(Treatment, abiomass, bbiomass)))
plant.sub.ab <- subset(data,  select=(c(Treatment, abiomass, bbiomass, soil)))
# Combining the above and belowground biomass # 
plant.m <- melt(plant.sub.ab) 

# Summarizing means of plant biomass data # 
plant.m.means <- ddply(plant.m, c("Treatment", "variable"), summarise, mean=mean(value))

# Plotting stacked bar graph # 
live.biomass.bar <- ggplot(plant.m.means, aes(x = variable, y = mean, fill = Treatment)) + 
  geom_bar(width=0.5, colour="black", position = position_dodge(width = 0.65), stat = "identity", size=1)

plant.m.sub <- subset(plant.m, variable == "bbiomass")


# Calculating standard error of means for total plant biomass #
plantmeans.sem <- ddply(plant.m, c("Treatment", "variable"), summarise,
                   mean=mean(value), sem=sd(value)/sqrt(length(value)))
plantmeans.sem <- transform(plantmeans.sem, lower=mean-sem, upper=mean+sem)

# Create label for axis.label.x # 
label <- c("Aboveground", "Belowground")

# Plotting stacked bargraph # 
live.biomass.bar  + geom_errorbar(aes(ymax=upper, ymin=lower),
                              position=position_dodge(0.65), data=plantmeans.sem, width = 0.2, size=1) +
  labs(x =  "Plant Section" ,y="Plant Biomass (g)") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Black")) +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =20),
        axis.title=element_text(size=22),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(15,10,0,10)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18))+
  annotate("text", x =0.55, y=1.1, label = c("B"), size=12) +
  scale_x_discrete(labels = label)




```


# 7) Does Rpf treatment affect total and belowground biomass? 

stacked bargraph of plant above and belowground biomass.
Two-factor ANOVA testing how soil type (sterlized versus live) and Rpf treatment affected belowground plant biomass. 

```{r}
# Total biomass # 
# Test of normality # 
qqnorm(plantfitness$total.biomass)
shapiro.test(data$total.biomass)
# Results: p = 0.06, data is normal # 

# Two-factor ANOVA of total biomass with treatment and soil type # 
plant.t.aov <- anova(lm(total.biomass ~ Treatment*soil, data=data))
plant.t.aov


live.soil <- subset(data, soil == "Live")
# Root:Shoot ratio
root.shoot.aov <- anova(lm(root.shoot.ratio ~ Treatment*soil, data=data))
root.shoot.aov


# Belowground biomass # 
# Two-factor ANOVA for belowground biomass changes with soil type and treatment # 
#bbiomass.aov <- anova(lm(bbiomass ~ Treatment, data=plantfitness))
#bbiomass.aov
# One-factor ANOVA for belowground biomass changes with treatment # 
bbiomass.aov <- aov(bbiomass ~ Treatment*soil, data=data)
summary(bbiomass.aov)
TukeyHSD(bbiomass.aov)

abiomass.aov <- aov(abiomass ~ Treatment*soil, data=data)
summary(abiomass.aov)

```


# 8) Visualization of microbial activity in soil 

 - 24 and 48 hour soil CO2 respiration taken each week over the course of growth chamber experiment. 
```{r}
# Set up #
rm(list = ls())
getwd()
setwd("~/../Github/BrassicaRpf/data")

# Require or install packages #
package.list <- c('plyr', 'grid' ,'png', 'car', 'bbmle', 'ggplot2', 'vegan', 'tidyr', 'dplyr', 'codyn', 'cowplot', 'MullerPlot', 'RColorBrewer', 'pander', 'reshape', 'nlme') 
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
  } }

# Load small function # 
sem <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}

fancy_scientific <- function(l) {
     # turn in to character string in scientific notation
     l <- format(l, scientific = TRUE)
     # quote the part before the exponent to keep all the digits
     l <- gsub("^(.*)e", "'\\1'e", l)
     # turn the 'e+' into plotmath format
     l <- gsub("e", "%*%10^", l)
     # return this as an expression
     parse(text=l)
}
```


# 3) Load data and calculate corrected copy number 

qPCR.csv : Quantitative PCR results of bacterial (EUB 338f-515r) and fungal (5.8f-ITs4r) gene copy abundances in live soil DNA taken from experimental pots containing Brassica plants. CSV file contains Treatment, Rep, Week, Well position, and Gene copy amount.

GCH_CO2.csv : CO2 respiration measurement of growth chamber soil samples for all six weeks. Contains sample ID, treatment (rpf+ = 1, rpf-= 2), soil (simple = 1, complex = 2), plant (present = 1, not present = 2), hour, week 1-6.

```{r , message=FALSE, warning=FALSE}
# Set working directory # 
setwd("~/../Github/BrassicaRpf/data")

# Read csv files # 
abundance <- read.csv("qPCR.txt", header=T) 
# Log transformed and standardized based on soil gram and extracted DNA concentration # 

CO2 <- read.csv("GCH_CO2.csv", header=T)

# Define factor levels for bacterial and fungal qPCR data # 
abundance$Week<- factor(abundance$Week)
abundance <- na.omit(abundance)

soil <- abundance$Soil
dna <- abundance$DNA
abundance$stdGeneCopy <- (abundance$GeneCopy)/soil#/dna
abundance$log10stdGeneCopy <- log10(abundance$stdGeneCopy)

# Define factor levels for microbial soil respiration data # 
```


# Plotting soil respiration 

```{r}
setwd("~/../Github/BrassicaRpf/data")

# Subsetting relevant data from CO2 dataset # 
CO2.sub <- subset(CO2, plant=="present" & hour =="24" & soil=="live", select=c(Treatment, soil, Week.1, Week.2, Week.3, Week.4, Week.5, Week.6))

# Change column names # 
colnames(CO2.sub) <- c("Treatment", "soil", "1","2","3","4","5","6")

# Melt data into three columns of treatment, soil and weeks # 
CO2.m <- melt(CO2.sub)
CO2.m$variable <- as.factor(CO2.m$variable)

# Standardizing the y-axis units by 24 hours # 
CO2.values <- CO2.m$value
CO2.m$StdCO2 <- CO2.values/24

# Summarizing data # 
CO2.means <- ddply(CO2.m, c("Treatment", "variable"), summarise, mean=mean(StdCO2))

# Calculating standard error of the means # 
CO2.means.sem <- ddply(CO2.m, c("Treatment", "variable"), summarise,
                   mean=mean(StdCO2), sem=sd(StdCO2)/sqrt(length(StdCO2)))

CO2.means.sem <- transform(CO2.means.sem, lower=mean-sem, upper=mean+sem)

# Plot line graph of summarized data # 
resp.line <- ggplot(CO2.means, aes(x=variable, y=mean, color=Treatment, linetype=)) + 
    geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=.075) +
    geom_line() +
    geom_point()

# The errorbars overlapped, so use position_dodge to move them horizontally
pd <- position_dodge(0) # move them .05 to the left and right

# Plot line graph # 
ggplot(CO2.means, aes(x=variable, y=mean, colour=Treatment, group=Treatment)) + 
  geom_errorbar(aes(ymax=upper, ymin=lower), position=position_dodge(0), 
                data=CO2.means.sem, width = 0.2, size=1) +
    geom_line(position=pd, size=2) +
    geom_point(position=pd, size=6, shape=21, fill="white") + # 21 is filled circle
    xlab("Time (Weeks)") +
    ylab(expression(Soil~Respiration~(ppm~CO[2]~d^-1))) +
    theme_classic() +
    theme(axis.text.y=element_text(colour="black", size=18),
          axis.text.x=element_text(colour="black", size=20),
          axis.title.y = element_text(size = 19, colour="black",  margin = margin(0,10,0,0)),
          axis.title.x = element_text(size = 20,colour="black",margin=margin(15,10,0,10)),
          panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA)) +
  scale_color_manual(values=c('Grey','Black')) + 
  theme(legend.position="none")

```


# 9) Time series analysis of CO2 respiration across growth chamber experiment 

Perform RM-ANOVA on CO2 respiration dataset. 

```{r}
# Perform RM-ANOVA #   
CO2.rm <- lme(value ~ variable*Treatment, random = ~ 1 | soil,
              correlation = corAR1(form = ~1 | soil), 
              data = CO2.m)

rich.cmp <- lme(value ~ variable*Treatment, random = ~ 1 | soil,
                correlation = corCompSymm(form = ~1 | soil),
                data = CO2.m)

# Next steps: Fix up CO2.m table so that week numbers are factors # 

# Look at detailed output #
#summary(rich.cmp)
#summary(CO2.rm)

# Make cleaner ANOVA table #
set.caption("RMANOVA for soil CO2 respiration")
pander(anova(rich.cmp))

# Make cleaner ANOVA table #
set.caption("RMANOVA for soil CO2 respiration")
pander(anova(CO2.rm))

# Multiple comparision tests # 
Rpf.CO2.aov <- aov(value ~ variable*Treatment, data=CO2.m)
summary(Rpf.CO2.aov)
TukeyHSD(Rpf.CO2.aov)

pairwise.t.test(Rpf.CO2.aov, ses, p.adj = "none")


require(multcomp)
summary(glht(rich.cmp, linfct=mcp(Treatment = "Tukey")), test = adjusted(type = "bonferroni"))
```

# 4) Visualization of microbial abundance in soil

qPCR gene copy abundances as a metric for bacterial (16s rRNA). 
Two-factor ANOVA testing the effect of Rpf treatment and week number (1 or 6) on bacterial soil abundance.
qPCR gene copy abundances as a metric for fungal (ITS) abundance 
Two-factor ANOVA testing the effect of Rpf treatment and week number (1 or 6) on fungal soil abundance.

## Bacterial gene copy abundance standardized by soil amount and DNA extracted

```{r}
# Subset bacterial 16S dataset # 
bac.abundance <- subset(abundance, Gene=="16S rRNA" , select=c(Treatment, Week, stdGeneCopy))

# Bacterial abundance: 16S rRNA gene copy abundance # 
bac.means <- ddply(bac.abundance, c("Treatment", "Week"), summarise, mean=mean(stdGeneCopy))

# Barplot of bacterial gene abundance in soil # 
bac.means.barplot <- ggplot(bac.means, aes(x=Week,y = mean, fill=Treatment)) + 
  geom_bar(width=0.5, colour="black", position = position_dodge(width = 0.65), stat = "identity", size=1) 

# Calculate bacterial gene abundance standard errors of the means # 
bac.means.sem <- ddply(bac.abundance, c("Treatment", "Week"), summarise,
                   mean=mean(stdGeneCopy), sem=sd(stdGeneCopy)/sqrt(length(stdGeneCopy)))
bac.means.sem <- transform(bac.means.sem, lower=mean-sem, upper=mean+sem)

# Barplot of bacterial gene abundance in soil for pub figures # 
bac.means.barplot + geom_errorbar(aes(ymax=upper, ymin=lower),
                              position=position_dodge(0.65), data=bac.means.sem, width = 0.2, size=1) +
  labs(x="Time (Weeks)", y="16S rRNA Gene Copy/Soil (g)") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Black")) +
  theme(axis.text.y=element_text(colour="black", size=16),
        axis.text.x=element_text(colour="black", size =20),
        axis.title.y = element_text(size = 19, colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(size = 20, colour="black",margin=margin(15,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18))+
  scale_y_continuous(labels=fancy_scientific)  +
  annotate("text", x = 0.6, y = 1750000, label = c("A"), size=12) 

```

## Fungal ITS gene copy abundance standardized by soil amount and DNA extracted

```{r}
# Fungal ITS # 
# Subset ITS gene copy data # 
ITS.abundance <- subset(abundance, Gene=="ITS" , select=c(Treatment, Week, stdGeneCopy))

# Fungal abundance: ITS gene copy abundance # 
ITS.means <- ddply(ITS.abundance, c("Treatment", "Week"), summarise, mean=mean(stdGeneCopy))

# Barplot of Fungal ITS abundance in soil # 
ITS.mean.barplot<- ggplot(ITS.means, aes(x=Week,y = mean, fill=Treatment)) +
  geom_bar(width=0.5, colour="black", position = position_dodge(width = 0.65), stat = "identity", size=1) 

# Calculate fungal ITS gene abundance standard errors of the means # 
ITS.means.sem <- ddply(ITS.abundance, c("Treatment", "Week"), summarise,
                   mean=mean(stdGeneCopy), sem=sd(stdGeneCopy)/sqrt(length(stdGeneCopy)))

ITS.means.sem <- transform(ITS.means.sem, lower=mean-sem, upper=mean+sem)

# Barplot of fungal gene abundance in soil for pub figures # 
ITS.mean.barplot + 
  geom_errorbar(aes(ymax=upper, ymin=lower), position=position_dodge(0.65), 
                data=ITS.means.sem, width = 0.2, size=1) +
  labs(x="Time (Weeks)", y="ITS Gene Copy/Soil (g)") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Black")) +
  theme(axis.text.y=element_text(colour="black", size=16),
        axis.text.x=element_text(colour="black", size =20),
        axis.title.y = element_text(size = 19, colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(size = 20, colour="black",margin=margin(15,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18)) +
  scale_y_continuous(labels=fancy_scientific) +
  annotate("text", x =0.6, y=2500000, label = c("B"), size=12)
```


# 5) Does Rpf treatment affect Bacterial and fungal gene copy abundances?

```{r}
# Bacterial 16S rRNA # 
# Two Factor ANOVA of 16S rRNA bacterial gene copy abundances affected by Treatment and week of growth chamber experiment # 
bac.aov <- aov(stdGeneCopy ~ Treatment*Week, data=bac.abundance)
summary(bac.aov)
TukeyHSD(bac.aov)

# Results: Rpf application to potted soil significantly (p<0.005) reduces bacterial 16S rRNA gene copy numbers # 
# Gene copy number increases significantly over the 6 week growth chamber experiment # 
bac.aov <- aov(stdGeneCopy ~ Treatment*Week, data=bac.abundance)
summary(bac.aov)
TukeyHSD(bac.aov)

fun.aov <- aov(stdGeneCopy ~ Treatment*Week, data=ITS.abundance)
summary(fun.aov)

```



# 1) Question: How does Rpf treatment affect soil microbial alpha and beta diversity?


```{r, message=FALSE, warning=FALSE}
# Setup work enviroment 
rm(list = ls())
setwd("C:/Users/vkuo/GitHub/BrassicaRpf/analysis")

# Source code functions # 
source("C:/Users/vkuo/GitHub/BrassicaRpf/bin/DiversityFunctions.R")
source("C:/Users/vkuo/GitHub/BrassicaRpf/bin/MothurTools.R")
source("C:/Users/vkuo/GitHub/BrassicaRpf/bin/phylodiversity2.R")

# Load dependencies # 
package.list <- c('vegan', 'plyr' , 'ggplot2', 'car', 'grid', 'png', 'ape', 'picante', 'ade4', 'phytools', 'phangorn', 'indicspecies', 'viridis' ,  'BiodiversityR') 
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
  } }

# Small custom functions # 
sem <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}

ttest <- function(reg, coefnum, val){
  co <- coef(summary(reg))
  tstat <- (co[coefnum,1]-val)/co[coefnum,2]
  pstat <- 2 * pt(abs(tstat), reg$df.residual, lower.tail = FALSE)
  return(list = c(t = tstat, df = reg$df.residual, p =  pstat))
}
```

# 3) Describe and load data set

.shared and .taxonomy file are outputs of alignment and clustering programing mothur that was run in IU mason. 

```{r}
## Define Inputs
# Design = general design file for experiment
# shared = OTU table from mothur with sequence similarity clustering
# tax = Taxonomy for 97% similarity OTUs

design <- "C:/Users/vkuo/GitHub/BrassicaRpf/data/Brassica.design.txt"
shared <- "C:/Users/vkuo/GitHub/BrassicaRpf/mothur/output/Brassica.bac.final.shared"
tax <- "C:/Users/vkuo/GitHub/BrassicaRpf/mothur/output/Brassica.bac.final.0.03.taxonomy"

# Run All: Select if all section are to be re-run
run.all <- TRUE
```

## Import Design, Shared, and Taxonomy files

```{r}
# Import Design
design <- read.delim(design, header=T, row.names=1)

# Import Shared Files
OTU <- read.otu(shared = shared, cutoff = "0.03")         # 97% Similarity

# Import Taxonomy
#OTU.tax <- read.tax(taxonomy = tax, format = "rdp")

#OTU.tre <- read.tree("../phylo/Brassica.bac.rename.tree.1")
```

# 4) Calculate Coverage Stats

```{r}
# Remove OTUs with less than two occurences across all sites # 
OTU <- OTU[, which(colSums(OTU) >= 2)]

# Remove mock community # 
OTU <- OTU[1:20, ]

# Determine coverage of sequences # 
cov.seqs <- count.groups(OTU)
cov.mean <- mean(cov.seqs) # 1160871
cov.sem <- sem(cov.seqs) # 16095
cov.min <- min(cov.seqs) # 79797
total.seqs <- sum(cov.seqs) # 3,217,419

# Good's coverage
goods.c <- function(x = ""){
              1 - (apply(OTU, 1, function(x){sum(x == 1)}) / rowSums(x))
}

goods.c.Brassica <- goods.c(OTU)
mean.good.c <- mean(goods.c.Brassica) # 0.98
min.good.c <- min(goods.c.Brassica) # 0.96 
# Coverage is good # 
```

## Calculate species richness

```{r}
# Species observation function
S.obs <- function(x = ""){
  rowSums(x > 0) * 1
}
  

# Chao1 function
S.chao1 <- function(x = ""){
  S.obs(x) + (sum(x == 1)^2) / (2 * sum(x == 2))
}

df <- as.matrix(S.chao1(OTU))

```


# 5) Alpha diversity

## Calculate Taxonomic Alpha diversity using Resampling

```{r}
if (run.all == TRUE){
  rich <- round(richness.iter(input = OTU, size = 45000,
                              iters = 100, shared = "FALSE"), 3)
  even <- round(evenness.iter(input = OTU, size = 45000,
                              iters = 100, shared = "FALSE",
                              method = "simp_even"), 3)

  rare <- rarefy(OTU, 45000, se = FALSE, MARGIN = 1)

  # Write output to files
  write.table(rich, "../data/rich.txt", sep = "\t",
              col.names = T, row.names = T)
  write.table(even, "../data/even.txt", sep = "\t",
              col.names = T, row.names = T)
}

# Read in alpha diversity files from above
rich2 <- read.table("../data/rich.txt", sep = "\t")
even2 <- read.table("../data/even.txt", sep = "\t")

# Merge data to design and calculate mean and sem per sample
rich.data <- merge(design, rich2, by = "row.names")
row.names(rich.data) <- rich.data$Row.names
rich.data <- rich.data[sort(row.names(rich.data)), ]
rich.mean <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, mean, na.rm = TRUE),3)
rich.sem <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, sem, na.rm = TRUE), 3)

even.data <- merge(design, even2, by = "row.names")
row.names(even.data) <- even.data$Row.names
even.data <- even.data[sort(row.names(even.data)), ]
even.mean <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, mean, na.rm = TRUE),3)
even.sem <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, sem, na.rm = TRUE),4)

# Make new dataframe merging design file and mean diversity
Brassica.div <- data.frame(design[sort(row.names(design)), ], rich.mean, even.mean)

# Take averages of technial reps
rich.rep.ave <- ddply(Brassica.div, .(treatment, type, rep.num), summarize, rich = mean(rich.mean))
even.rep.ave <- ddply(Brassica.div, .(treatment, type, rep.num), summarize, even = mean(even.mean))

# Reshape data
rich.2 <- reshape(rich.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep.num"), direction = "wide")

even.2 <- reshape(even.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep.num"), direction = "wide")
# Results: 
# Mean richness for active and Rpf+ communities is 3833.306, evenness is 0.050
# Mean richness for active and Rpf- communities is 3430.956, evenness is 0.0596
# Mean richness for total and Rpf+ communities is 2886.790, evenness is 0.059
# Mean richness for total and Rpf- communities is 3236.336, evenness is 0.0564
```

## Richness barplot 

```{r}
# Calculate biomass means and standard errors of the means # 
rich.cDNA.means.sem <- ddply(rich.2, c("treatment"), summarise,
                   mean=mean(rich.cDNA), sem=sd(rich.cDNA)/sqrt(length(rich.cDNA)))
rich.cDNA.means.sem <- transform(rich.cDNA.means.sem, lower=mean-sem, upper=mean+sem)

rich.DNA.means.sem <- ddply(rich.2, c("treatment"), summarise,
                   mean=mean(rich.DNA), sem=sd(rich.DNA)/sqrt(length(rich.DNA)))
rich.DNA.means.sem <- transform(rich.DNA.means.sem, lower=mean-sem, upper=mean+sem)

rich.means <- rbind(rich.DNA.means.sem,rich.cDNA.means.sem)
rich.means$Community <- c("DNA", "DNA", "cDNA", "cDNA")

# Bargraph #
rich.barplot<- ggplot(rich.means, aes(x=Community,y = mean, fill=treatment)) + 
  geom_bar(width=0.5,colour="black", position = position_dodge(width = 0.65), stat = "identity", size=1)  

# Clean Bargraph figure #
rich.barplot + geom_errorbar(aes(ymax=upper, ymin=lower),
                              position=position_dodge(0.65), data=rich.means, width = 0.2, size=1) +
  labs(x="Community Type", y="Richness") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Black")) +
  theme(axis.text.y=element_text(colour="black", size=16),
        axis.text.x=element_text(colour="black", size =14),
        axis.title=element_text(size=18),
        axis.title.y = element_text(colour="black",  margin = margin(0,15,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(10,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=1.5, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 1.5),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18))
```

## Richness: differences among treatment?

```{r}
# Rpf treatment to soil # 
rich.anova.c <- aov(rich.2$rich.cDNA ~ rich.2$treatment)
summary(rich.anova.c)
TukeyHSD(rich.anova.c)

rich.anova.d <- aov(rich.2$rich.DNA ~ rich.2$treatment)
summary(rich.anova.d)
TukeyHSD(rich.anova.d)
# Results: 
# Active community richness was not significantly affected by Rpf treatment p=0.479, evenness p=0.316
# Total community richness was not significantly affected by Rpf treatment p=0.33, evenness p=0.865
```



# 6) Beta Diversity

## Taxonomic Beta Diversity 

```{r}
# Make presence-absence matrix
OTU.PA <- (OTU > 0) * 1

# Make relative abundence matrix
OTU.REL <- OTU
for (i in 1:dim(OTU)[1]){
  OTU.REL[i,] <- OTU[i,]/sum(OTU[i,])
  }

# Log-transform relative abundances
OTU.REL.log <- decostand(OTU, method="log")

Brassica.bc.dis <- vegdist(OTU.REL.log, method = "bray", binary = "FALSE")
Brassica.dis.mean <- mean(Brassica.bc.dis)

# Principal Coordinates Analysis (PCoA)
Brassica.PCoA <- cmdscale(Brassica.bc.dis, eig = TRUE, k = 3)
explainvar1 <- round(Brassica.PCoA$eig[1] / sum(Brassica.PCoA$eig), 3) * 100
explainvar2 <- round(Brassica.PCoA$eig[2] / sum(Brassica.PCoA$eig), 3) * 100
explainvar3 <- round(Brassica.PCoA$eig[3] / sum(Brassica.PCoA$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

# OTU Scores
otu.scores <- t(cor(Brassica.PCoA$points, OTU.REL))
otu.scores <- as.matrix(otu.scores)[,1:2]
otu.scores <- otu.scores[abs(otu.scores[,1]) > 0.7|abs(otu.scores[,2]) > 0.7,]

# Average BC Distance Between Treatments
Brassica.bc.dis.m <- as.matrix(Brassica.bc.dis)
all.equal(row.names(Brassica.div), rownames(Brassica.bc.dis.m))

treatment.div <- unique(Brassica.div$treatment)
treatment.dis <- rep(NA, length(treatment.div))
for(i in 1:length(treatment.div)){
  temp <- row.names(Brassica.div[Brassica.div$treatment == treatment.div[i], ])
  treatment.dis[i] <- Brassica.bc.dis.m[temp[1], temp[2]]
}

mean(treatment.dis)  # The mean bray curtis dis site similarity is 0.477
```

## PCoA plots

```{r}
#png(filename="../figures/FigureS2-Ordinations.png",
#    width = 1800, height = 800, res = 96*2)
layout(matrix(1:3, 1, 3), widths = c(20, 9, 2.5))
par(mar = c(7, 10, 1, 0) + 0.5)

plot(Brassica.PCoA$points[ ,1], Brassica.PCoA$points[ ,2],
     ylim = c(-0.4, 0.4), xlim = c(-0.5, 0.4),
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""), line= 5,
     #xlab = "", ylab = "", xaxt = "n", yaxt = "n",
     pch = 22, cex = 2.0, type = "n", cex.lab = 2.5, cex.axis = 3,
     axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 2, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 1, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 2, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

#mtext(side = 3, "Taxonomic PCoA", line = 1.5, cex = 1)
#text("a", x = -0.48, y = 0.38, font = 2, cex = 1.8)

# Subset data
Brassica.div.sort <- Brassica.div[order(Brassica.div[,1]) ,]
all.equal(row.names(Brassica.PCoA$points), rownames(Brassica.div.sort))
Brassica.points <- data.frame(Brassica.PCoA$points, Brassica.div.sort)

# Active community
Brassica.active.rpf <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.active.no <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf-"), ]
# Total community
Brassica.total.rpf <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.total.no <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf-"), ]

# Add points
# Active community Rpf+
points(Brassica.active.rpf[ ,1], Brassica.active.rpf[ ,2], pch = 21,
       cex = 3, col = "black", bg = "Black")
# Active community Rpf-
points(Brassica.active.no[ ,1], Brassica.active.no[ ,2], pch = 21,
       cex = 3, col = "black", bg = "White")   
# Total community Rpf+ 
points(Brassica.total.rpf[ ,1], Brassica.total.rpf[ ,2], pch = 22,
       cex = 3, col = "black", bg = "Black")
# Total community Rpf-
points(Brassica.total.no[ ,1], Brassica.total.no[ ,2], pch = 22,
       cex = 3, col = "black", bg = "White")

# Add Legend Outside
par(mar = c(4, 0, 5, 1) + 0.5)
plot.new()
legend(0, 1, c("Active Rpf-", "Active Rpf+", "Total Rpf-", "Total Rpf+"),
       pch = c(21, 21, 22, 22),
       pt.bg = c("White", "Black", "White", "Black"),
       bty = "n", y.intersp = 1, pt.cex = 3, cex = 2)

```

# Beta diversity hypothesis testing 

## PERMANOVA: Taxonomic

```{r}
# Check Order of Dataframes
all.equal(row.names(Brassica.div), row.names(OTU.REL.log))

# Run PERMANOVA with adonis function # 
Brassica.permanova <- adonis(OTU.REL.log ~ Brassica.div$type * Brassica.div$treatment,
                         method = "bray", binary = FALSE, permutations = 9999)
Brassica.permanova

Brassica.bc.dis <- vegdist(OTU.REL.log, method = "bray", binary = "FALSE")
beta.disp <- betadisper(d = Brassica.bc.dis, group = Brassica.div$treatment)
permutest(beta.disp, 999)
TukeyHSD(beta.disp)
# Results: Activity status affects the microbial community compositon --> expected
# Rpf treatment affect, but not signifcantly, the microbial community (p=0.089) 

```




