---
title: "Soil Bacterial Community"
author: "Venus Kuo"
date: "March 1, 2018"
output: html_document
---

# 1) Question

How does Rpf treatment affect soil microbial alpha and beta diversity?


# 2) Set working environment and load packages

I sequenced the 16S rRNA gene of soil bacterial DNA and cDNA to determine how Rpf treatment affected the soil bacterial community richness, diveristy, and composition at the end of the growth chamber experiment. 


# 3) Set up work environment and load dependencies 

```{r, message=FALSE, warning=FALSE}
# Setup work enviroment 
rm(list = ls())
setwd("~/../Github/BrassicaRpf/analysis")

# Source code functions # 
source("C:/Users/vkuo/GitHub/BrassicaRpf/bin/DiversityFunctions.R")
source("C:/Users/vkuo/GitHub/BrassicaRpf/bin/MothurTools.R")
source("C:/Users/vkuo/GitHub/BrassicaRpf/bin/phylodiversity2.R")

# Load dependencies # 
package.list <- c('vegan', 'plyr' , 'ggplot2', 'car', 'grid', 'png', 'ape', 'picante', 'ade4', 'phytools', 'phangorn', 'indicspecies', 'viridis' ,  'BiodiversityR') 
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
  } }

# Load small custom functions # 
sem <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}

ttest <- function(reg, coefnum, val){
  co <- coef(summary(reg))
  tstat <- (co[coefnum,1]-val)/co[coefnum,2]
  pstat <- 2 * pt(abs(tstat), reg$df.residual, lower.tail = FALSE)
  return(list = c(t = tstat, df = reg$df.residual, p =  pstat))
}
```

# 3) Describe and load data set

design = general design file for experiment
shared = OTU table from mothur with sequence similarity clustering
tax = Taxonomy for 97% similarity OTUs

## Define inputs

```{r}
design <- "C:/Users/vkuo/GitHub/BrassicaRpf/data/Brassica.design.txt"
shared <- "C:/Users/vkuo/GitHub/BrassicaRpf/mothur/output/Brassica.bac.final.shared"
tax <- "C:/Users/vkuo/GitHub/BrassicaRpf/mothur/output/Brassica.bac.final.0.03.taxonomy"

# Run All: Select if all section are to be re-run
run.all <- TRUE
```

## Import Design, Shared, and Taxonomy files

```{r}
# Import Design
design <- read.delim(design, header=T, row.names=1)

# Import Shared Files
OTU <- read.otu(shared = shared, cutoff = "0.03")         # 97% Similarity

# Import Taxonomy
OTU.tax <- read.tax(taxonomy = tax, format = "rdp")

OTU.tre <- read.tree("../phylo/Brassica.bac.rename.tree.2")
```


# 4) Calculate Coverage Stats

```{r}
# Remove OTUs with less than two occurences across all sites # 
OTU <- OTU[, which(colSums(OTU) >= 2)]

# Remove mock community # 
OTU <- OTU[1:20, ]   # Only 20 total samples

# Determine coverage of sequences # 
cov.seqs <- count.groups(OTU)
cov.mean <- mean(cov.seqs) # 160,871
cov.sem <- sem(cov.seqs) # 16,095.38
cov.min <- min(cov.seqs) # 79,797
total.seqs <- sum(cov.seqs) # 3,217,419

# Good's coverage
goods.c <- function(x = ""){
              1 - (apply(OTU, 1, function(x){sum(x == 1)}) / rowSums(x))
}

goods.c.Brassica <- goods.c(OTU)
mean.good.c <- mean(goods.c.Brassica) # 0.984  Good mean coverage
min.good.c <- min(goods.c.Brassica) # 0.967  Good lowest coverage
```


# 5) Alpha diversity

## Calculate Taxonomic Alpha diversity using Resampling

```{r}
# Mario's resampling code to estimate alpha diversity (used if run.all = T)
if (run.all == TRUE){
  rich <- round(richness.iter(input = OTU, size = 10000,
                              iters = 100, shared = "FALSE"), 3)
  even <- round(evenness.iter(input = OTU, size = 10000,
                              iters = 100, shared = "FALSE",
                              method = "simp_even"), 3)
  rare <- rarefy(OTU, 10000, se = FALSE, MARGIN = 1)
  # Write output to files
  write.table(rich, "../data/rich.txt", sep = "\t",
              col.names = T, row.names = T)
  write.table(even, "../data/even.txt", sep = "\t",
              col.names = T, row.names = T)
}

# Read in alpha diversity files from above
rich2 <- read.table("../data/rich.txt", sep = "\t")
even2 <- read.table("../data/even.txt", sep = "\t")

# Merge data to design and calculate mean and sem per sample
rich.data <- merge(design, rich2, by = "row.names")
row.names(rich.data) <- rich.data$Row.names
rich.data <- rich.data[sort(row.names(rich.data)), ]
rich.mean <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, mean, na.rm = TRUE),3)
rich.sem <- round(apply(rich.data[5:(4 + dim(rich2)[2])], 1, sem, na.rm = TRUE), 3)

even.data <- merge(design, even2, by = "row.names")
row.names(even.data) <- even.data$Row.names
even.data <- even.data[sort(row.names(even.data)), ]
even.mean <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, mean, na.rm = TRUE),3)
even.sem <- round(apply(even.data[5:(4 + dim(even2)[2])], 1, sem, na.rm = TRUE),4)

# Make new dataframe merging design file and mean diversity
Brassica.div <- data.frame(design[sort(row.names(design)), ], rich.mean, even.mean)

# Take averages of technial reps
rich.rep.ave <- ddply(Brassica.div, .(treatment, type, rep.num), summarize, rich = mean(rich.mean))
even.rep.ave <- ddply(Brassica.div, .(treatment, type, rep.num), summarize, even = mean(even.mean))

# Reshape data 
rich.2 <- reshape(rich.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep.num"), direction = "wide")

even.2 <- reshape(even.rep.ave[,1:4], timevar = "type",
                   idvar = c("treatment", "rep.num"), direction = "wide")
```

## Richness barplot 

```{r}
# Calculate biomass means and standard errors of the means # 
rich.cDNA.means.sem <- ddply(rich.2, c("treatment"), summarise,
                   mean=mean(rich.cDNA), sem=sd(rich.cDNA)/sqrt(length(rich.cDNA)))
rich.cDNA.means.sem <- transform(rich.cDNA.means.sem, lower=mean-sem, upper=mean+sem)

rich.DNA.means.sem <- ddply(rich.2, c("treatment"), summarise,
                   mean=mean(rich.DNA), sem=sd(rich.DNA)/sqrt(length(rich.DNA)))
rich.DNA.means.sem <- transform(rich.DNA.means.sem, lower=mean-sem, upper=mean+sem)

rich.means <- rbind(rich.DNA.means.sem,rich.cDNA.means.sem)
rich.means$Community <- c("DNA", "DNA", "cDNA", "cDNA")

# Bacterial richness barplot #
rich.barplot<- ggplot(rich.means, aes(x=Community,y = mean, fill=treatment)) + 
  geom_bar(width=0.5,colour="black", position = position_dodge(width = 0.65), stat = "identity", size=1.5)  

rich.barplot + 
  geom_errorbar(aes(ymax=upper, ymin=lower), position=position_dodge(0.65), 
                data=rich.means, width = 0.2, size=1) +
  labs(x="Community Type", y="Richness") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Grey")) +
  theme(axis.text.y=element_text(colour="black", size=16),
        axis.text.x=element_text(colour="black", size =14),
        axis.title=element_text(size=18),
        axis.title.y = element_text(colour="black",  margin = margin(0,15,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(10,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=1.5, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 1.5),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18))

# Results: 
# Mean richness for active and Rpf+ communities is 2039.166, evenness is 0.050
# Mean richness for active and Rpf- communities is 1911.550, evenness is 0.0596
# Mean richness for total and Rpf+ communities is 1763.540, evenness is 0.059
# Mean richness for total and Rpf- communities is 1923.586, evenness is 0.0564
```

## Richness: differences among treatment?

```{r}
# Rpf treatment to soil # 
rich.anova.c <- aov(rich.mean ~ treatment*type, Brassica.div)
summary(rich.anova.c)   
#TukeyHSD(rich.anova.c)
# Results: No significant effect of metabolic status (p=0.38) or Rpf (p=0.9) or the interaction (p=0.3)
```


## Evenness barplot 

```{r}
# Calculate biomass means and standard errors of the means # 
even.cDNA.means.sem <- ddply(even.2, c("treatment"), summarise,
                   mean=mean(even.cDNA), sem=sd(even.cDNA)/sqrt(length(even.cDNA)))
even.cDNA.means.sem <- transform(even.cDNA.means.sem, lower=mean-sem, upper=mean+sem)

even.DNA.means.sem <- ddply(even.2, c("treatment"), summarise,
                   mean=mean(even.DNA), sem=sd(even.DNA)/sqrt(length(even.DNA)))
even.DNA.means.sem <- transform(even.DNA.means.sem, lower=mean-sem, upper=mean+sem)

even.means <- rbind(even.DNA.means.sem,even.cDNA.means.sem)
even.means$Community <- c("DNA", "DNA", "cDNA", "cDNA")

# Bargraph #
even.barplot<- ggplot(even.means, aes(x=Community,y = mean, fill=treatment)) + 
  geom_bar(width=0.5,colour="black", position = position_dodge(width = 0.65), stat = "identity", size=1.5)  

# Clean Bargraph figure #
even.barplot + geom_errorbar(aes(ymax=upper, ymin=lower),
                              position=position_dodge(0.65), data=even.means, width = 0.2, size=1) +
  labs(x="Community Type", y="Evenness") +
  theme_classic() +
  scale_fill_manual(values=c("White", "Grey")) +
  theme(axis.text.y=element_text(colour="black", size=16),
        axis.text.x=element_text(colour="black", size =14),
        axis.title=element_text(size=18),
        axis.title.y = element_text(colour="black",  margin = margin(0,15,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(10,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=1.5, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 1.5),
        legend.title = element_text(size=18),
        legend.text=element_text(size=18))
```

## Evenness: differences among sites?

```{r}
even.anova.c <- aov(even.mean ~ treatment*type, Brassica.div)
summary(even.anova.c)
#TukeyHSD(even.anova.c)
# Results: No significant effect of Rpf (p=0.7) or metabolic status (p=0.7) or interaction (0.6)
```


# 6) Beta Diversity

## Taxonomic Beta Diversity 

```{r}
# Make presence-absence matrix
OTU.PA <- (OTU > 0) * 1

# Make relative abundence matrix
OTU.REL <- OTU
for (i in 1:dim(OTU)[1]){
  OTU.REL[i,] <- OTU[i,]/sum(OTU[i,])
  }

# Log-transform relative abundances
OTU.REL.log <- decostand(OTU, method="log")

Brassica.bc.dis <- vegdist(OTU.REL.log, method = "bray", binary = "FALSE")
Brassica.dis.mean <- mean(Brassica.bc.dis)

# Principal Coordinates Analysis (PCoA)
Brassica.PCoA <- cmdscale(Brassica.bc.dis, eig = TRUE, k = 3)
explainvar1 <- round(Brassica.PCoA$eig[1] / sum(Brassica.PCoA$eig), 3) * 100
explainvar2 <- round(Brassica.PCoA$eig[2] / sum(Brassica.PCoA$eig), 3) * 100
explainvar3 <- round(Brassica.PCoA$eig[3] / sum(Brassica.PCoA$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

# OTU Scores
otu.scores <- t(cor(Brassica.PCoA$points, OTU.REL))
otu.scores <- as.matrix(otu.scores)[,1:2]
otu.scores <- otu.scores[abs(otu.scores[,1]) > 0.7|abs(otu.scores[,2]) > 0.7,]

# Average BC Distance Between Treatments
Brassica.bc.dis.m <- as.matrix(Brassica.bc.dis)
all.equal(row.names(Brassica.div), rownames(Brassica.bc.dis.m))

treatment.div <- unique(Brassica.div$treatment)
treatment.dis <- rep(NA, length(treatment.div))
for(i in 1:length(treatment.div)){
  temp <- row.names(Brassica.div[Brassica.div$treatment == treatment.div[i], ])
  treatment.dis[i] <- Brassica.bc.dis.m[temp[1], temp[2]]
}

mean(treatment.dis)  # The mean bray curtis dis site similarity is 0.477
```

## PCoA plots

```{r}
# Plot figure # 
png(filename="../figures/FigureS4-Ordination.png",
    width = 1800, height = 800, res = 96*2)

layout(matrix(1:3, 1, 3), widths = c(20, 9, 2.5))

par(mar = c(7, 10, 1, 0) + 0.5)

plot(Brassica.PCoA$points[ ,1], Brassica.PCoA$points[ ,2],
     ylim = c(-0.4, 0.4), xlim = c(-0.5, 0.4),
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""), line= 5,
     #xlab = "", ylab = "", xaxt = "n", yaxt = "n",
     pch = 22, cex = 2.0, type = "n", cex.lab = 2.5, cex.axis = 3,
     axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 2, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 1, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 2, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Subset data
Brassica.div.sort <- Brassica.div[order(Brassica.div[,1]) ,]
all.equal(row.names(Brassica.PCoA$points), rownames(Brassica.div.sort))
Brassica.points <- data.frame(Brassica.PCoA$points, Brassica.div.sort)

# Active community
Brassica.active.rpf <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.active.no <- Brassica.points[ which(Brassica.points$type == "cDNA" &
                                   Brassica.points$treatment == "Rpf-"), ]
# Total community
Brassica.total.rpf <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf+"), ]
Brassica.total.no <- Brassica.points[ which(Brassica.points$type == "DNA" &
                                   Brassica.points$treatment == "Rpf-"), ]

# Add points
# Active community Rpf+
points(Brassica.active.rpf[ ,1], Brassica.active.rpf[ ,2], pch = 21,
       cex = 3.5, col = "black", bg = "Black", lwd= 2.5)
# Active community Rpf-
points(Brassica.active.no[ ,1], Brassica.active.no[ ,2], pch = 21,
       cex = 3.5, col = "black", bg = "White", lwd= 2.5)   
# Total community Rpf+ 
points(Brassica.total.rpf[ ,1], Brassica.total.rpf[ ,2], pch = 22,
       cex = 3.5, col = "black", bg = "Black", lwd= 2.5)
# Total community Rpf-
points(Brassica.total.no[ ,1], Brassica.total.no[ ,2], pch = 22,
       cex = 3.5, col = "black", bg = "White", lwd= 2.5)

# Add Legend Outside
par(mar = c(4, 0, 5, 1) + 0.5)
plot.new()
legend(0, 1, c("Active Rpf-", "Active Rpf+", "Total Rpf-", "Total Rpf+"),
       pch = c(21, 21, 22, 22),
       pt.bg = c("White", "Black", "White", "Black"),
       bty = "n", y.intersp = 1, pt.cex = 3.5, cex = 2, lwd= 2.5, lty =  NA)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/Figure5-Ordination.png")
grid.raster(img)
```

# Beta diversity hypothesis testing 

## PERMANOVA: Taxonomic

```{r}
# Check Order of Dataframes
all.equal(row.names(Brassica.div), row.names(OTU.REL.log))

# Run PERMANOVA with adonis function # 
Brassica.permanova <- adonis(OTU.REL.log ~ Brassica.div$type * Brassica.div$treatment,
                         method = "bray", binary = FALSE, permutations = 999)
Brassica.permanova
# Results: Significant effect of metabolic status (r2 = 0.225, p=0.001)
# Marginally non-significant effect of Rpf (r=0.061, p=0.092)
# Rpf treatment affect, but not signifcantly, the microbial community (p=0.089) 
```





